// @adyen/adyen-platform-experience-web/dist/es/components/internal/FormFields/Select/BalanceAccountSelector/BalanceAccountSelector.js
import { jsx as s } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { memo as n } from "../../../../../external/preact/compat/dist/compat.module.js";
import { useCallback as a } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import { useResponsiveViewport as C, mediaQueries as p } from "../../../../../hooks/useResponsiveViewport.js";
import { renderDefaultSingleSelectionCheckedness as c } from "../components/SelectListItem.js";
import "./BalanceAccountSelector.scss.js";
import { Fragment as d } from "../../../../../external/preact/dist/preact.module.js";
import S from "../Select.js";
const m = "adyen-pe-balance-account-selector", _ = `${m}__account-id`, f = `${m}__account-label`, u = n(
  ({
    activeBalanceAccount: r,
    balanceAccountSelectionOptions: o,
    onBalanceAccountSelection: t
  }) => {
    const l = C(p.down.xs), i = a(
      (e) => /* @__PURE__ */ s(d, { children: [
        /* @__PURE__ */ s("div", { className: e.contentClassName, children: [
          e.item.name && /* @__PURE__ */ s("span", { className: f, children: e.item.name }),
          /* @__PURE__ */ s("span", { className: _, children: e.item.id })
        ] }),
        c(e)
      ] }),
      []
    );
    return o && o.length > 1 ? /* @__PURE__ */ s(
      S,
      {
        popoverClassNameModifiers: [m],
        onChange: t,
        filterable: !1,
        multiSelect: !1,
        selected: r == null ? void 0 : r.id,
        withoutCollapseIndicator: !0,
        items: o,
        renderListItem: i,
        showOverlay: l
      }
    ) : null;
  }
), O = u;
export {
  O as default
};
