// @adyen/adyen-platform-experience-web/dist/es/components/external/TransactionsOverview/components/TransactionsTable/PaymentMethodCell.js
import { jsx as o } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import m from "../../../../../core/Context/useCoreContext.js";
import { Image as s } from "../../../../internal/Image/Image.js";
import { Tag as t } from "../../../../internal/Tag/Tag.js";
import { TagVariant as l } from "../../../../internal/Tag/types.js";
import { TypographyVariant as p } from "../../../../internal/Typography/types.js";
import T from "../../../../internal/Typography/Typography.js";
import { useResponsiveViewport as f, mediaQueries as n } from "../../../../../hooks/useResponsiveViewport.js";
import { parsePaymentMethodType as _ } from "../utils.js";
import { PAYMENT_METHOD_CLASS as g, PAYMENT_METHOD_LOGO_CONTAINER_CLASS as N, PAYMENT_METHOD_LOGO_CLASS as O } from "./constants.js";
import { Fragment as C } from "../../../../../external/preact/dist/preact.module.js";
const E = ({
  paymentMethod: r,
  bankAccount: e
}) => {
  const { i18n: i } = m(), a = f(n.down.xs);
  return /* @__PURE__ */ o("div", { className: g, children: r || e ? /* @__PURE__ */ o(C, { children: [
    /* @__PURE__ */ o("div", { className: N, children: /* @__PURE__ */ o(
      s,
      {
        name: r ? r.type : "bankTransfer",
        alt: r ? r.type : "bankTransfer",
        folder: "logos/",
        className: O
      }
    ) }),
    /* @__PURE__ */ o(T, { variant: p.BODY, stronger: a, children: r ? _(r) : e == null ? void 0 : e.accountNumberLastFourDigits })
  ] }) : /* @__PURE__ */ o(t, { label: i.get("noData"), variant: l.LIGHT_WITH_OUTLINE }) });
}, V = E;
export {
  V as default
};
