// @adyen/adyen-platform-experience-web/dist/es/components/internal/DataOverviewDisplay/DataDetailsModal.js
import { jsx as r } from "../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { useEffect as u } from "../../../external/preact/hooks/dist/hooks.module.js";
import { popoverUtil as h } from "../Popover/utils/popoverUtil.js";
import C from "../../../core/Context/useCoreContext.js";
import M from "../Modal/ModalContent/ModalContent.js";
import g from "../Modal/Modal.js";
const O = ({
  children: p,
  className: f,
  selectedDetail: o,
  resetDetails: i
}) => {
  const { i18n: m } = C(), n = !!o;
  return u(() => {
    n && h.closeAll();
  }, [n]), /* @__PURE__ */ r("div", { className: f, children: [
    p,
    o && /* @__PURE__ */ r(
      g,
      {
        title: o != null && o.title ? m.get(o.title) : void 0,
        isOpen: !!o,
        "aria-label": m.get("payoutDetails"),
        onClose: i,
        isDismissible: !0,
        headerWithBorder: !1,
        size: (o == null ? void 0 : o.modalSize) ?? "large",
        children: o && /* @__PURE__ */ r(M, { ...o == null ? void 0 : o.selection })
      }
    )
  ] });
};
export {
  O as DataDetailsModal
};
