// @adyen/adyen-platform-experience-web/dist/es/components/internal/Alert/Alert.js
import { jsx as e } from "../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import d from "classnames";
import { ButtonVariant as y } from "../Button/types.js";
import { TypographyElement as m, TypographyVariant as a } from "../Typography/types.js";
import p from "../Typography/Typography.js";
import { AlertIcon as f } from "./AlertIcon.js";
import { AlertVariantOption as t } from "./types.js";
import "./Alert.scss.js";
import u from "../Button/Button.js";
import T from "../Icon/Icon.js";
const _ = ({ className: c, description: o, title: n, type: l, children: s, onClose: i, variant: r = t.DEFAULT }) => /* @__PURE__ */ e("div", { className: d("adyen-pe-alert", `adyen-pe-alert--${l}`, `adyen-pe-alert--${r}`, c), role: "alert", children: [
  /* @__PURE__ */ e(f, { type: l, className: "adyen-pe-alert__icon" }),
  /* @__PURE__ */ e("div", { className: "adyen-pe-alert__content", children: [
    n && r !== t.TIP && /* @__PURE__ */ e(p, { className: "adyen-pe-alert__title", el: m.DIV, variant: a.BODY, wide: !0, strongest: !0, children: n }),
    o && /* @__PURE__ */ e(
      p,
      {
        className: "adyen-pe-alert__description",
        el: m.DIV,
        variant: r !== t.TIP ? a.CAPTION : a.BODY,
        wide: !0,
        children: o
      }
    ),
    s
  ] }),
  i && r !== t.TIP && /* @__PURE__ */ e("div", { className: "adyen-pe-alert__close-button", children: /* @__PURE__ */ e(u, { iconButton: !0, variant: y.TERTIARY, onClick: i, children: /* @__PURE__ */ e(T, { name: "cross" }) }) })
] }), v = _;
export {
  _ as Alert,
  v as default
};
