// @adyen/adyen-platform-experience-web/dist/es/components/internal/FilterBar/FilterBar.js
import { jsx as i } from "../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import l from "classnames";
import { ButtonVariant as n } from "../Button/types.js";
import { useState as s, useEffect as c } from "../../../external/preact/hooks/dist/hooks.module.js";
import m from "../../../core/Context/useCoreContext.js";
import { useResponsiveViewport as u, mediaQueries as f } from "../../../hooks/useResponsiveViewport.js";
import "./FilterBar.scss.js";
import a from "../Button/Button.js";
import d from "../Icon/Icon.js";
import { isFunction as b } from "../../../utils/value/is.js";
const o = "adyen-pe-filter-bar-mobile-switch", E = () => {
  const t = u(f.down.xs), [e, r] = s(!t);
  return c(() => {
    r(!t);
  }, [t]), { isMobileViewport: t, showingFilters: e, setShowingFilters: r };
}, I = ({ isMobileViewport: t, showingFilters: e, setShowingFilters: r }) => t ? /* @__PURE__ */ i("div", { className: o, children: /* @__PURE__ */ i(
  a,
  {
    iconButton: !0,
    className: `${o}__button`,
    disabled: !b(r),
    onClick: () => r == null ? void 0 : r(!e),
    variant: n.SECONDARY,
    children: /* @__PURE__ */ i(d, { name: e ? "cross" : "filter" })
  }
) }) : null, B = (t) => {
  const { i18n: e } = m();
  return t.showingFilters ? /* @__PURE__ */ i(
    "div",
    {
      "aria-label": e.get("filterBar"),
      className: l("adyen-pe-filter-bar", { "adyen-pe-filter-bar__content--mobile": t.isMobileViewport }),
      children: [
        t.children,
        t.canResetFilters && !!t.resetFilters && /* @__PURE__ */ i(a, { variant: n.TERTIARY, onClick: t.resetFilters, children: e.get("button.clearAll") })
      ]
    }
  ) : null;
}, M = B;
export {
  B as FilterBar,
  I as FilterBarMobileSwitch,
  M as default,
  E as useFilterBarState
};
