// @adyen/adyen-platform-experience-web/dist/es/components/internal/Button/DownloadButton/DownloadButton.js
import { jsx as o } from "../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import g from "classnames";
import { useState as k, useEffect as m } from "../../../../external/preact/hooks/dist/hooks.module.js";
import x from "../../../../core/Context/useCoreContext.js";
import { useResponsiveViewport as B, mediaQueries as y } from "../../../../hooks/useResponsiveViewport.js";
import h from "../../SVGIcons/Download.js";
import v from "../Button.js";
import { ButtonVariant as b } from "../types.js";
import C from "./useDownload.js";
import "./DownloadButton.scss.js";
import L from "../../Spinner/Spinner.js";
function E({ blob: r, filename: c }) {
  const t = document.createElement("a"), n = URL.createObjectURL(r);
  t.href = n, t.download = c || "download";
  const a = () => {
    setTimeout(() => {
      URL.revokeObjectURL(n);
    }, 150);
  };
  t.addEventListener("click", a, { once: !0 }), t.click();
}
function Y({ className: r, disabled: c, endpointName: t, params: n, setError: a, errorDisplay: f, onDownloadRequested: d }) {
  const { i18n: p } = x(), [s, u] = k(!1), _ = B(y.down.xs), { data: l, error: i, isFetching: e } = C(t, n, s);
  m(() => {
    s && u(!1);
  }, [s]), m(() => {
    l && E(l);
  }, [l]), m(() => {
    a && i && a(i);
  }, [i]);
  const w = () => {
    u(!0), d == null || d();
  };
  return /* @__PURE__ */ o("div", { className: "adyen-pe-download", children: [
    _ ? /* @__PURE__ */ o(v, { iconButton: !0, variant: b.TERTIARY, onClick: w, children: e ? /* @__PURE__ */ o(L, { size: "small" }) : /* @__PURE__ */ o(h, {}) }) : /* @__PURE__ */ o(
      v,
      {
        className: g("adyen-pe-download__button", { "adyen-pe-download__button--loading": e }, r),
        disabled: c || e,
        variant: b.SECONDARY,
        onClick: w,
        iconLeft: e ? /* @__PURE__ */ o(L, { size: "small" }) : /* @__PURE__ */ o(h, {}),
        children: e ? `${p.get("downloading")}..` : p.get("download")
      }
    ),
    i && f && /* @__PURE__ */ o("div", { className: "adyen-pe-download__error", children: f })
  ] });
}
export {
  Y as default
};
