// @adyen/adyen-platform-experience-web/dist/es/components/internal/DataOverviewContainer/DataOverviewContainer.js
import { jsx as e } from "../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { WITH_ERROR_CLASS as u } from "./constants.js";
import { ErrorMessageDisplay as t } from "../ErrorMessageDisplay/ErrorMessageDisplay.js";
import c from "classnames";
import { getErrorMessage as g } from "../../utils/getErrorMessage.js";
import { Fragment as h } from "../../../external/preact/dist/preact.module.js";
import { useAuthContext as f } from "../../../core/Auth/context.js";
function x({
  balanceAccountsError: r,
  children: a,
  className: n,
  errorMessage: o,
  isBalanceAccountIdWrong: i,
  onContactSupport: s
}) {
  const { hasError: m } = f();
  return /* @__PURE__ */ e("div", { className: c(n, { [u]: m }), children: m ? /* @__PURE__ */ e(
    t,
    {
      withImage: !0,
      centered: !0,
      title: "somethingWentWrong",
      message: [o, "tryRefreshingThePageOrComeBackLater"],
      refreshComponent: !0
    }
  ) : r ? /* @__PURE__ */ e(
    t,
    {
      withImage: !0,
      centered: !0,
      ...g(r, "weCouldNotLoadYourBalanceAccounts", s)
    }
  ) : i ? /* @__PURE__ */ e(
    t,
    {
      withImage: !0,
      centered: !0,
      title: "somethingWentWrong",
      message: [o, "theSelectedBalanceAccountIsIncorrect"]
    }
  ) : /* @__PURE__ */ e(h, { children: a }) });
}
export {
  x as default
};
