// @adyen/adyen-platform-experience-web/dist/es/components/internal/ExpandableCard/ExpandableCard.js
import { jsx as e } from "../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import s from "classnames";
import { useState as v, useCallback as h, useRef as l, useEffect as I } from "../../../external/preact/hooks/dist/hooks.module.js";
import "./ExpandableCard.scss.js";
import b from "../SVGIcons/ChevronUp.js";
import B from "../SVGIcons/ChevronDown.js";
import { BASE_CLASS as D, CONTAINER_CLASS as m, CONTAINER_BUTTON_CLASS as L, CONTAINER_FILLED_CLASS as p, CONTAINER_OVERLAY_ID as C, CONTENT_CLASS as N, CONTENT_EXPANDABLE_CLASS as x, CHEVRON_CLASS as R, CONTAINER_OVERLAY_CLASS as g, CONTAINER_HIDDEN_CLASS as k } from "./constants.js";
import y from "../../../core/Context/useCoreContext.js";
import { useClickOutside as F } from "../../../hooks/element/useClickOutside.js";
import { Fragment as V } from "../../../external/preact/dist/preact.module.js";
import T from "../BaseButton/BaseButton.js";
const U = ({ renderHeader: t, children: u, filled: o, fullWidth: f, ...i }) => {
  const { i18n: A } = y(), [a, S] = v(!1), r = h(() => S((n) => !n), [S]), E = l(null), c = l(!1), O = l(a), d = F(
    void 0,
    h(() => {
      a && (r(), c.current = !0);
    }, [a, r])
  );
  return I(() => {
    var n, _;
    a ? (n = d.current) == null || n.focus() : (O.current !== a && !c.current && ((_ = E.current) == null || _.focus()), c.current = !1), O.current = a;
  }, [a, d]), /* @__PURE__ */ e("div", { className: D, children: u ? /* @__PURE__ */ e(V, { children: [
    /* @__PURE__ */ e(
      T,
      {
        className: s(m, L, { [p]: o }),
        disabled: a,
        fullWidth: f,
        "aria-controls": C,
        "aria-expanded": a,
        "aria-hidden": a,
        onClick: r,
        ref: E,
        "data-testid": "expand-button",
        ...i,
        children: [
          /* @__PURE__ */ e("span", { className: "adyen-pe-sr-only", children: A.get("expandableCard.expand") }),
          /* @__PURE__ */ e("div", { className: s(N, x), children: t }),
          /* @__PURE__ */ e("div", { className: R, children: /* @__PURE__ */ e(B, { role: "presentation" }) })
        ]
      }
    ),
    /* @__PURE__ */ e(
      T,
      {
        id: C,
        className: s(m, L, g, {
          [p]: o,
          [k]: !a
        }),
        disabled: !a,
        fullWidth: f,
        "aria-controls": C,
        "aria-expanded": a,
        "aria-hidden": !a,
        onClick: r,
        ref: d,
        "data-testid": "collapse-button",
        ...i,
        children: [
          /* @__PURE__ */ e("span", { className: "adyen-pe-sr-only", children: A.get("expandableCard.collapse") }),
          /* @__PURE__ */ e("div", { className: s(N, x), children: [
            t,
            /* @__PURE__ */ e("div", { children: u })
          ] }),
          /* @__PURE__ */ e("div", { className: R, children: /* @__PURE__ */ e(b, { role: "presentation" }) })
        ]
      }
    )
  ] }) : /* @__PURE__ */ e("div", { className: s(m, { [p]: o }), ...i, children: /* @__PURE__ */ e("div", { className: N, children: t }) }) });
}, K = U;
export {
  K as default
};
