// @adyen/adyen-platform-experience-web/dist/es/components/internal/Card/Card.js
import { jsx as s } from "../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { CARD_BASE_CLASS as O, CARD_FILLED as h, CARD_NO_OUTLINE as I, CARD_NO_PADDING as L, CARD_HEADER as n, CARD_HEADER_CONTENT as p, CARD_TITLE as d, CARD_SUBTITLE as t, CARD_BODY as f, CARD_BODY_WITH_TITLE as B, CARD_FOOTER as S } from "./constants.js";
import C from "classnames";
import "./Card.scss.js";
const v = ({
  title: c,
  subTitle: _,
  children: m,
  footer: A,
  el: R,
  renderHeader: a,
  renderFooter: D,
  filled: N,
  noOutline: E,
  noPadding: i,
  classNameModifiers: T,
  testId: l
}) => {
  const o = R || "header";
  return /* @__PURE__ */ s(
    "section",
    {
      className: C(
        O,
        { [h]: N, [I]: E, [L]: i },
        T
      ),
      "data-testid": l,
      children: [
        (c || a) && /* @__PURE__ */ s(o, { className: n, children: (c || a) && /* @__PURE__ */ s("div", { className: p, children: [
          a || /* @__PURE__ */ s("span", { className: d, children: c }),
          _ && /* @__PURE__ */ s("div", { className: t, children: _ })
        ] }) }),
        /* @__PURE__ */ s(
          "div",
          {
            className: C(f, {
              [B]: c || a
            }),
            children: m
          }
        ),
        (A || D) && /* @__PURE__ */ s("footer", { className: S, children: D || A })
      ]
    }
  );
}, g = v;
export {
  g as default
};
