// @adyen/adyen-platform-experience-web/dist/es/components/internal/Tabs/Tabs.js
import { jsx as n } from "../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { useState as u, useRef as y, useCallback as h, useMemo as T } from "../../../external/preact/hooks/dist/hooks.module.js";
import "./Tabs.scss.js";
import _ from "../../../core/Context/useCoreContext.js";
import g from "../Typography/Typography.js";
import { TypographyVariant as I, TypographyElement as N } from "../Typography/types.js";
import { InteractionKeyCode as r } from "../../types.js";
function O(l) {
  const s = l.tabs, { i18n: d } = _(), c = l.defaultActiveTab ? l.tabs.findIndex((e) => e.id === l.defaultActiveTab) : 0, [t, b] = u(c === -1 ? 0 : c), m = y({}), o = h((e) => {
    const a = m.current[e];
    a == null || a.focus();
  }, []), p = T(() => {
    const e = s.length, a = {
      [r.ARROW_RIGHT]: () => o((t + 1) % e),
      [r.ARROW_LEFT]: () => o((t - 1 + e) % e),
      [r.HOME]: () => o(0),
      [r.END]: () => o(e - 1)
    };
    return (i) => {
      var f;
      a[i.key] && (i.preventDefault(), (f = a[i.key]) == null || f.call(a));
    };
  }, [s.length, t, o]);
  return /* @__PURE__ */ n("section", { "aria-label": d.get("tabs"), children: [
    /* @__PURE__ */ n("div", { className: "adyen-fp-tabs", role: "tablist", "aria-orientation": "horizontal", children: s.map((e, a) => /* @__PURE__ */ n(
      "button",
      {
        id: `tab-id-${e.id}`,
        name: e.id,
        className: "adyen-fp-tabs__tab",
        role: "tab",
        "aria-controls": `panel-id-${e.id}`,
        "aria-selected": t === a,
        ref: (i) => m.current[a] = i,
        onKeyDown: p,
        onClick: a === t ? void 0 : () => b(a),
        onFocus: () => b(a),
        tabIndex: t === a ? 0 : -1,
        disabled: e.disabled,
        children: /* @__PURE__ */ n(
          g,
          {
            variant: I.BODY,
            el: N.SPAN,
            stronger: !0,
            className: "adyen-fp-tabs__tab-content-tab-label",
            children: d.get(e.label)
          }
        )
      },
      `tab-${e.id}`
    )) }),
    /* @__PURE__ */ n("div", { className: "adyen-fp-tabpanel__wrapper", children: s.map((e, a) => /* @__PURE__ */ n(
      "section",
      {
        name: e.id,
        className: "adyen-fp-tabpanel__content",
        id: `panel-id-${e.id}`,
        hidden: t !== a,
        role: "tabpanel",
        "aria-labelledby": `tab-id-${e.id}`,
        children: e.content
      },
      `tabpanel-${e.id}`
    )) })
  ] });
}
export {
  O as default
};
