// @adyen/adyen-platform-experience-web/dist/es/components/internal/Popover/PopoverDismissButton/PopoverDismissButton.js
import { jsx as o } from "../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { ButtonVariant as n } from "../../Button/types.js";
import "./PopoverDismissButton.scss.js";
import r from "../../../../core/Context/useCoreContext.js";
import { memo as s } from "../../../../external/preact/compat/dist/compat.module.js";
import l from "../../Button/Button.js";
import { Fragment as m } from "../../../../external/preact/dist/preact.module.js";
function a({ image: t = !0, onClick: e }) {
  const { i18n: i } = r();
  return /* @__PURE__ */ o(m, { children: /* @__PURE__ */ o(
    l,
    {
      className: (() => t ? "adyen-pe-popover-dismiss-button--on-image" : "")(),
      iconButton: !0,
      variant: n.TERTIARY,
      onClick: e,
      "aria-label": i.get("closeIconLabel"),
      children: /* @__PURE__ */ o("svg", { role: "img", xmlns: "http://www.w3.org/2000/svg", width: "16", height: "16", fill: "none", children: [
        /* @__PURE__ */ o("title", { children: "dismiss" }),
        /* @__PURE__ */ o(
          "path",
          {
            fill: "#00112C",
            fillRule: "evenodd",
            d: "M11.4697 12.5303C11.7626 12.8232 12.2374 12.8232 12.5303 12.5303C12.8232 12.2374 12.8232 11.7626 12.5303 11.4697L9.06066 8L12.5303 4.53033C12.8232 4.23744 12.8232 3.76256 12.5303 3.46967C12.2374 3.17678 11.7626 3.17678 11.4697 3.46967L8 6.93934L4.53033 3.46967C4.23744 3.17678 3.76256 3.17678 3.46967 3.46967C3.17678 3.76256 3.17678 4.23744 3.46967 4.53033L6.93934 8L3.46967 11.4697C3.17678 11.7626 3.17678 12.2374 3.46967 12.5303C3.76256 12.8232 4.23744 12.8232 4.53033 12.5303L8 9.06066L11.4697 12.5303Z",
            clipRule: "evenodd"
          }
        )
      ] })
    }
  ) });
}
const L = s(a);
export {
  L as default
};
