// @adyen/adyen-platform-experience-web/dist/es/components/internal/Popover/Popover.js
import { jsx as r } from "../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import oo from "../Button/ButtonActions/ButtonActions.js";
import { ButtonActionsLayoutBasic as ro } from "../Button/ButtonActions/types.js";
import { DEFAULT_POPOVER_CLASSNAME as c, DEFAULT_TOOLTIP_CLASSNAME as eo, POPOVER_CONTAINER_CLASSNAME as no, TOOLTIP_CONTENT_CLASSNAME as U, POPOVER_CONTENT_CLASSNAME as N, POPOVER_HEADER_TITLE_CLASSNAME as to, POPOVER_FOOTER_CLASSNAME as co, POPOVER_HEADER_CLASSNAME as B } from "./constants.js";
import io from "./PopoverDismissButton/PopoverDismissButton.js";
import so from "./PopoverTitle/PopoverTitle.js";
import { PopoverContainerVariant as i, PopoverContainerSize as k } from "./types.js";
import { InteractionKeyCode as uo } from "../../types.js";
import { useClickOutside as mo, ClickOutsideVariant as po, CONTROL_ELEMENT_PROPERTY as O } from "../../../hooks/element/useClickOutside.js";
import ao from "../../../hooks/element/useFocusTrap.js";
import fo from "../../../hooks/element/usePopoverPositioner.js";
import w from "../../../hooks/element/useUniqueIdentifier.js";
import lo from "../../../hooks/useReflex.js";
import { SELECTORS as Oo, isFocusable as Eo } from "../../../primitives/dom/tabbableRoot/tabbable.js";
import b from "classnames";
import { createPortal as Po } from "../../../external/preact/compat/dist/compat.module.js";
import { useMemo as q, useRef as L, useCallback as C, useEffect as x } from "../../../external/preact/hooks/dist/hooks.module.js";
import "./Popover.scss.js";
import { Fragment as _ } from "../../../external/preact/dist/preact.module.js";
import { isFunction as To } from "../../../utils/value/is.js";
import { boolOrTrue as Ao } from "../../../utils/value/bool.js";
import { getModifierClasses as No } from "../../../utils/preact/className.js";
const Lo = (s) => {
  var e;
  let p;
  const l = (e = s.querySelector(`.${U}`)) == null ? void 0 : e.querySelectorAll(Oo);
  return l ? (Array.prototype.some.call(l, (u) => {
    if (Eo(u))
      return p = u;
  }), p) : null;
}, Co = (s) => s === i.TOOLTIP ? [10, 3, 5, 5] : [15, 15, 15, 15];
function jo({
  actions: s,
  disableFocusTrap: p = !1,
  actionsLayout: l = ro.SPACE_BETWEEN,
  variant: e = i.TOOLTIP,
  title: u,
  open: a,
  dismissible: y,
  modifiers: K,
  divider: S,
  fitContent: R,
  withoutSpace: v,
  containerSize: d,
  position: W,
  targetElement: o,
  setToTargetWidth: H,
  dismiss: n,
  children: I,
  withContentPadding: j,
  classNameModifiers: G,
  showOverlay: f = !1,
  fitPosition: Y,
  ...F
}) {
  const M = q(() => To(n) && Ao(y), [n, y]), V = w(), $ = L(), J = C(
    (t) => {
      var m;
      n && n(), t && ((m = o == null ? void 0 : o.current) == null || m.focus());
    },
    [n, o]
  ), E = C(
    (t) => {
      t.code === uo.ESCAPE && (n && n(), (o == null ? void 0 : o.current).focus());
    },
    [n, o]
  ), P = L(E), h = L(), D = mo(
    fo(Co(e), o, e, W, V, H, f, Y),
    n,
    e === i.TOOLTIP && !a,
    po.POPOVER
  ), Q = ao(p ? null : D, J), T = lo(
    C(
      (t, m) => {
        m instanceof Element && (m[O] = void 0, delete m[O]), t instanceof Element && (t[O] = o.current, cancelAnimationFrame(h.current), h.current = requestAnimationFrame(() => {
          if ($.current === a || !($.current = a))
            return;
          const A = Lo(t);
          A == null || A.focus();
        }));
      },
      [a, o]
    ),
    p ? D : Q
  ), X = w(T), Z = q(
    () => ({
      [`${c}--medium`]: d === k.MEDIUM,
      [`${c}--with-divider`]: !!S,
      [`${c}--wide`]: d === k.WIDE,
      [`${c}--fit-content`]: R,
      [`${c}--without-space`]: v,
      [`${c}--auto-width`]: f
    }),
    [d, S, v, R, f]
  );
  x(() => {
    T.current && (T.current[O] = o.current);
  }, [o]), x(() => (document.removeEventListener("keydown", P.current), document.addEventListener("keydown", P.current = E), () => document.removeEventListener("keydown", P.current)), [E]);
  const z = e === i.TOOLTIP ? eo : `${c} ${no}`, g = e === i.TOOLTIP ? U : `${N}`;
  return Po(
    /* @__PURE__ */ r(_, { children: a ? /* @__PURE__ */ r(_, { children: [
      f && /* @__PURE__ */ r("div", { className: "adyen-pe-popover__overlay" }),
      /* @__PURE__ */ r(
        "div",
        {
          id: "popover",
          ref: X,
          ...F,
          className: b(z, Z, G),
          style: { visibility: "hidden" },
          role: F.role ?? (e === i.POPOVER ? "dialog" : "tooltip"),
          children: [
            (u || M) && /* @__PURE__ */ r("div", { className: No(B, K, [B]), children: [
              u && /* @__PURE__ */ r("div", { className: to, children: /* @__PURE__ */ r(so, { title: u }) }),
              M && /* @__PURE__ */ r(io, { onClick: n })
            ] }),
            I && /* @__PURE__ */ r(_, { children: [
              /* @__PURE__ */ r(
                "div",
                {
                  className: b(g, {
                    [`${N}--with-padding`]: j,
                    [`${N}--overlay`]: f
                  }),
                  children: I
                }
              ),
              e === i.TOOLTIP && /* @__PURE__ */ r("span", { "data-popover-placement": "hidden", ref: V, className: "adyen-pe-tooltip__arrow" })
            ] }),
            s && /* @__PURE__ */ r("div", { className: co, children: /* @__PURE__ */ r(oo, { actions: s, layout: l }) })
          ]
        }
      )
    ] }) : null }),
    document.getElementsByTagName("body")[0]
  );
}
export {
  jo as default
};
