// @adyen/adyen-platform-experience-web/dist/es/components/external/UIElement/UIElement.js
var m = Object.defineProperty;
var p = (o, e, t) => e in o ? m(o, e, { enumerable: !0, configurable: !0, writable: !0, value: t }) : o[e] = t;
var s = (o, e, t) => (p(o, typeof e != "symbol" ? e + "" : e, t), t);
import { jsx as i } from "../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import l from "../../../core/Context/CoreProvider.js";
import c from "../BaseElement.js";
import "./UIElement.scss.js";
import u from "classnames";
import { AuthProvider as h } from "../../../core/Auth/context.js";
class d extends c {
  constructor(t) {
    super(t);
    s(this, "componentRef", null);
    s(this, "componentToRender", null);
    s(this, "elementRef");
    s(this, "onContactSupport");
    s(this, "customClassNames");
    s(this, "setUIElementStatus");
    this.setState = this.setState.bind(this), this.onContactSupport = t.onContactSupport, this.elementRef = this;
  }
  get isValid() {
    return !!this.state.isValid;
  }
  /**
   * Get the element's displayable name
   */
  get displayName() {
    return this.type;
  }
  /**
   * Get the element accessible name, used in the aria-label of the button that controls selected component
   */
  get accessibleName() {
    return this.displayName;
  }
  /**
   * Return the type of an element
   */
  get type() {
    var t;
    return (t = this.constructor) == null ? void 0 : t.type;
  }
  formatProps(t) {
    return t;
  }
  /**
   * Formats the component data output
   */
  formatData() {
    return {
      ...this.state
    };
  }
  setState(t) {
    this.state = { ...this.state, ...t };
  }
  setStatus(t, n) {
    var r, a;
    return (r = this.componentRef) != null && r.setStatus ? this.componentRef.setStatus(t, n) : (a = this.setUIElementStatus) == null || a.call(this, t), this;
  }
  render() {
    const t = this.props.core, n = t.update.bind(t), r = this.props.onError || t.onError || null;
    return t.session.errorHandler = r, /* @__PURE__ */ i(h, { type: this.type, session: t.session, children: /* @__PURE__ */ i(
      l,
      {
        i18n: t.localization.i18n,
        loadingContext: t.loadingContext,
        updateCore: n,
        externalErrorHandler: r,
        children: this.componentToRender && /* @__PURE__ */ i("div", { className: u("adyen-pe-component", this.customClassNames), children: this.componentToRender() })
      }
    ) }, performance.now());
  }
}
const g = d;
export {
  d as UIElement,
  g as default
};
