// @adyen/adyen-platform-experience-web/dist/es/components/external/TransactionsOverview/components/TransactionTotals/TransactionTotals.js
import { jsx as s } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { useCallback as y, useMemo as q } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import { useFetch as L } from "../../../../../hooks/useFetch.js";
import { BASE_CLASS as M, ITEM_CLASS as F } from "./constants.js";
import { memo as N } from "../../../../../external/preact/compat/dist/compat.module.js";
import { useResponsiveViewport as _, mediaQueries as k } from "../../../../../hooks/useResponsiveViewport.js";
import "./TransactionTotals.scss.js";
import { TotalsCard as p } from "./TotalsCard.js";
import { Fragment as v } from "../../../../../external/preact/dist/preact.module.js";
import { useAuthContext as B } from "../../../../../core/Auth/context.js";
import { EMPTY_OBJECT as j } from "../../../../../utils/value/constants.js";
const w = N(
  ({
    availableCurrencies: i,
    isAvailableCurrenciesFetching: g,
    balanceAccountId: r,
    createdSince: f,
    createdUntil: d,
    categories: c,
    statuses: l,
    maxAmount: T,
    minAmount: h,
    currencies: C,
    fullWidth: n
  }) => {
    const { getTransactionTotals: t } = B().endpoints, x = y(async () => t == null ? void 0 : t(j, {
      query: {
        createdSince: f,
        createdUntil: d,
        categories: c,
        statuses: l,
        maxAmount: T,
        minAmount: h,
        currencies: C,
        balanceAccountId: r
      }
    }), [r, c, f, d, C, t, T, h, l]), { data: o, isFetching: S } = L({
      fetchOptions: q(() => ({ enabled: !!r && !!t }), [r, t]),
      queryFn: x
    }), m = !r || S || g, a = y(() => {
      if (!i || !o)
        return o == null ? void 0 : o.data;
      const u = i.map((e) => o.data.find((O) => O.currency === e) || { currency: e, incomings: 0, expenses: 0 });
      return u.concat(o.data.filter((e) => !u.includes(e)));
    }, [i, o])() ?? [], E = _(k.only.xs);
    return /* @__PURE__ */ s("div", { className: M, children: E ? /* @__PURE__ */ s(v, { children: [
      /* @__PURE__ */ s("div", { className: F, children: /* @__PURE__ */ s(p, { totals: a, isLoading: m, hiddenField: "expenses", fullWidth: n }) }),
      /* @__PURE__ */ s("div", { className: F, children: /* @__PURE__ */ s(p, { totals: a, isLoading: m, hiddenField: "incomings", fullWidth: n }) })
    ] }) : /* @__PURE__ */ s(p, { totals: a, isLoading: m, fullWidth: n }) });
  }
), Z = w;
export {
  Z as default
};
