// @adyen/adyen-platform-experience-web/dist/es/components/external/CapitalOverview/CapitalOverviewElement.js
var l = Object.defineProperty;
var h = (o, e, t) => e in o ? l(o, e, { enumerable: !0, configurable: !0, writable: !0, value: t }) : o[e] = t;
var n = (o, e, t) => (h(o, typeof e != "symbol" ? e + "" : e, t), t);
import { jsx as u } from "../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import f from "../UIElement/UIElement.js";
import { CapitalOverview as v } from "./components/CapitalOverview/CapitalOverview.js";
import w from "../../../core/Auth/session/utils/sessionReady.js";
import { EMPTY_OBJECT as c } from "../../../utils/value/constants.js";
import { noop as m } from "../../../utils/common.js";
class d extends f {
  constructor(t) {
    super(t);
    n(this, "componentToRender", () => /* @__PURE__ */ u(v, { ...this.props, ref: (t) => void (this.componentRef = t) }));
    this.customClassNames = "adyen-pe-capital-overview-component", this.componentToRender = this.componentToRender.bind(this);
  }
  async getState() {
    var p;
    const { session: t } = this.props.core;
    await w(t);
    const { getDynamicGrantOffersConfiguration: s, getGrants: i } = t.context.endpoints, [a, r] = await Promise.all([
      s == null ? void 0 : s(c).catch(m),
      i == null ? void 0 : i(c).catch(m)
    ]);
    return { state: r && ((p = r.data) == null ? void 0 : p.length) > 0 ? "hasRequestedGrants" : a && a.minAmount ? "isPreQualified" : "isUnqualified" };
  }
}
n(d, "type", "capitalOverview");
const q = d;
export {
  d as CapitalOverviewElement,
  q as default
};
