// @adyen/adyen-platform-experience-web/dist/es/components/internal/Calendar/calendar/timeframe/frames/MonthFrame.js
var rt = Object.defineProperty;
var ot = (r, s, t) => s in r ? rt(r, s, { enumerable: !0, configurable: !0, writable: !0, value: t }) : r[s] = t;
var Z = (r, s, t) => (ot(r, typeof s != "symbol" ? s + "" : s, t), t), $ = (r, s, t) => {
  if (!s.has(r))
    throw TypeError("Cannot " + t);
};
var i = (r, s, t) => ($(r, s, "read from private field"), t ? t.call(r) : s.get(r)), n = (r, s, t) => {
  if (s.has(r))
    throw TypeError("Cannot add the same private member more than once");
  s instanceof WeakSet ? s.add(r) : s.set(r, t);
}, u = (r, s, t, e) => ($(r, s, "write to private field"), e ? e.call(r, t) : s.set(r, t), t);
var c = (r, s, t) => ($(r, s, "access private method"), t);
import nt from "./TimeFrame.js";
import { getWeekendDays as X } from "../common/utils.js";
import { getEdgesDistance as j, withTimezone as q, getTimezoneDateParts as k, computeTimestampOffset as mt, getMonthDays as b, startOfMonth as at, startOfWeek as ht } from "../../utils.js";
import { DAY_OF_WEEK_FORMATS as ft, MAXIMUM_MONTH_UNITS as J, DAY_MS as Q } from "../../constants.js";
import ct from "../common/flags.js";
import { TimeFlag as m } from "../../types.js";
import { timezoneToSystem as V, systemToTimezone as tt } from "../../../../../../core/Localization/datetime/restamper/utils.js";
import { isUndefined as lt, isString as gt } from "../../../../../../utils/value/is.js";
import { isInfinity as ut, isBitSafeInteger as pt } from "../../../../../../utils/value/number.js";
import { enumerable as K } from "../../../../../../utils/struct/property.js";
import { struct as _, withFreezeProxyHandlers as Tt, structFrom as dt } from "../../../../../../utils/struct/main.js";
var y, D, S, M, E, O, R, F, H, N, B, W, L, Y, w, P, T, I, z, v;
class At extends nt {
  constructor() {
    super();
    n(this, L);
    n(this, w);
    n(this, T);
    n(this, z);
    n(this, y, 7);
    n(this, D, []);
    n(this, S, X(this.firstWeekDay));
    n(this, M, void 0);
    n(this, E, -1 / 0);
    n(this, O, 1 / 0);
    n(this, R, 1 / 0);
    n(this, F, void 0);
    n(this, H, void 0);
    n(this, N, void 0);
    n(this, B, void 0);
    n(this, W, void 0);
    Z(this, "daysInWeek", i(this, y));
    this.initialize();
  }
  get fromTimestamp() {
    return i(this, E);
  }
  get toTimestamp() {
    return i(this, O);
  }
  get numberOfBlocks() {
    return i(this, R);
  }
  get currentDayTimestamp() {
    return i(this, M);
  }
  get dynamicBlockHeight() {
    return super.dynamicBlockHeight;
  }
  set dynamicBlockHeight(t) {
    const e = this.dynamicBlockHeight;
    super.dynamicBlockHeight = t, this.dynamicBlockHeight !== e && this.refreshFrame(!0);
  }
  get rowspan() {
    return i(this, y);
  }
  get firstWeekDay() {
    return super.firstWeekDay;
  }
  set firstWeekDay(t) {
    const e = this.firstWeekDay;
    super.firstWeekDay = t, this.firstWeekDay !== e && (i(this, D).length = 0, u(this, S, X(this.firstWeekDay)), this.reoriginate(), this.refreshFrame());
  }
  getCursorBlockOriginTimestampOffset(t) {
    return k(t, this.timezone)[2] - 1;
  }
  getDayOfWeekAtIndex(t) {
    if (!i(this, D)[t]) {
      const e = new Date(this.getTimestampAtIndex(t));
      let o = 0;
      i(this, S).includes(t) && (o |= m.WEEKEND), t === 0 ? o |= m.LINE_START : t === 6 && (o |= m.LINE_END);
      const l = {};
      for (const f of ft)
        l[f] = K(
          e.toLocaleDateString(this.locale, { weekday: f, timeZone: this.timezone })
        );
      i(this, D)[t] = _({
        flags: K(ct(o)),
        labels: K(_(l))
      });
    }
    return i(this, D)[t];
  }
  getEdgeBlockOffsetsFromOrigin() {
    return [c(this, L, Y).call(this, i(this, E)), c(this, L, Y).call(this, i(this, O))];
  }
  getFormattedDataForBlockCell(t) {
    const [e, o, l] = k(t, this.timezone);
    return [Number(l).toLocaleString(this.locale), `${e}-${`${o + 1}`.padStart(2, "0")}-${`${l}`.padStart(2, "0")}`];
  }
  getFormattedDataForFrameBlock(t) {
    const [e, o] = k(t, this.timezone);
    return [
      new Date(t).toLocaleDateString(this.locale, { month: "long", year: "numeric", timeZone: this.timezone }),
      `${e}-${`${o + 1}`.padStart(2, "0")}`
    ];
  }
  getFrameBlockAtIndex(t) {
    const [e, o, l] = b(this.origin, i(this, N), t), f = t > 0 ? this.getFrameBlockAtIndex(t - 1).inner.to + 1 : i(this, F), a = f + e - 1, g = Math.floor(f / 7) * 7, A = this.dynamicBlockHeight ? Math.ceil((a + 1) / 7) * 7 : g + J, G = this.dynamicBlockHeight ? A - g : J, et = new Proxy(
      _(),
      Tt({
        get: (it, x, st) => {
          if (gt(x)) {
            const C = +x;
            if (pt(C) && C >= 0 && C < G) {
              const d = g + C, p = this.getTimestampAtIndex(d), U = d % i(this, y);
              let h = p === this.currentDayTimestamp ? m.CURRENT : 0;
              return d === this.cursor && (h |= m.CURSOR), i(this, S).includes(U) && (h |= m.WEEKEND), U === 0 ? h |= m.LINE_START : U === i(this, y) - 1 && (h |= m.LINE_END), d >= f && d <= a && (d === f ? h |= m.BLOCK_START : d === a && (h |= m.BLOCK_END), h |= m.WITHIN_BLOCK), p >= this.fromTimestamp && p <= this.toTimestamp && (p === this.fromTimestamp && (h |= m.RANGE_START), p === this.toTimestamp && (h |= m.RANGE_END), h |= m.WITHIN_RANGE), p >= i(this, B) && p <= i(this, W) && (p === i(this, B) && (h |= m.SELECTION_START), p === i(this, W) && (h |= m.SELECTION_END), h |= m.WITHIN_SELECTION), [p, h];
            }
          }
          return Reflect.get(it, x, st);
        }
      })
    );
    return dt(et, {
      inner: {
        value: _({
          from: { value: f },
          to: { value: a },
          units: { value: e }
        })
      },
      month: { value: o },
      outer: {
        value: _({
          from: { value: g },
          to: { value: A - 1 },
          units: { value: G }
        })
      },
      year: { value: l }
    });
  }
  getUnitsForFrameBlockAtIndex(t) {
    return b(this.origin, i(this, N), t)[0];
  }
  getUnitsOffsetForTimestamp(t, e) {
    return Math.round((e - t) / Q);
  }
  reoriginate() {
    this.originTimestamp = at(this.originTimestamp, this.timezone);
    const [t, e] = k(this.originTimestamp, this.timezone), o = ht(this.originTimestamp, this.timezone, this.firstWeekDay);
    this.origin = e, u(this, N, t), u(this, F, this.getUnitsOffsetForTimestamp(o, this.originTimestamp)), u(this, H, c(this, w, P).call(this, this.originTimestamp, -i(this, F)));
  }
  reslice() {
    c(this, z, v).call(this), u(this, E, c(this, T, I).call(this, super.fromTimestamp)), u(this, O, c(this, T, I).call(this, super.toTimestamp)), u(this, R, j(super.fromTimestamp, super.toTimestamp, this.timezone) + 1);
  }
  shiftOrigin(t) {
    const [e, o] = k(this.originTimestamp, this.timezone), [, l, f] = b(o, e, t), a = q(this.timezone), g = new Date(V(a, this.originTimestamp)).setFullYear(f, l);
    this.originTimestamp = tt(a, g), this.reoriginate();
  }
  clearSelection() {
    super.clearSelection(), c(this, z, v).call(this), this.refreshFrame(!0);
  }
  getTimestampAtIndex(t) {
    return c(this, w, P).call(this, i(this, H), t);
  }
  updateSelection(t, e) {
    super.updateSelection(t, e), c(this, z, v).call(this), this.refreshFrame(!0);
  }
  withCurrentDayTimestamp() {
    u(this, M, c(this, T, I).call(this, Date.now()));
  }
}
y = new WeakMap(), D = new WeakMap(), S = new WeakMap(), M = new WeakMap(), E = new WeakMap(), O = new WeakMap(), R = new WeakMap(), F = new WeakMap(), H = new WeakMap(), N = new WeakMap(), B = new WeakMap(), W = new WeakMap(), L = new WeakSet(), Y = function(t) {
  const e = j(t, this.originTimestamp, this.timezone);
  return t < this.originTimestamp ? 0 - e : e;
}, w = new WeakSet(), P = function(t, e = 0) {
  const o = q(this.timezone), l = V(o, t), f = tt(o, l + e * Q);
  let [, , , a, g] = k(f, this.timezone), A = 0;
  return (a > 0 || g > 0) && (a = (a > 12 ? 24 : 0) - a, g = (a > 1 ? 1 : -1) * g, A = a * 36e5 + g * 6e4), f + A;
}, T = new WeakSet(), I = function(t) {
  return lt(t) || ut(t) ? t : t - mt(t, this.timezone);
}, z = new WeakSet(), v = function() {
  u(this, B, c(this, T, I).call(this, this.selectionStart)), u(this, W, c(this, T, I).call(this, this.selectionEnd));
};
export {
  At as default
};
