// @adyen/adyen-platform-experience-web/dist/es/components/internal/ProgressBar/ProgressBar.js
import { jsx as e } from "../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { useMemo as s } from "../../../external/preact/hooks/dist/hooks.module.js";
import u from "classnames";
import "./ProgressBar.scss.js";
import o from "../Typography/Typography.js";
import { TypographyElement as m, TypographyVariant as p } from "../Typography/types.js";
import { Tooltip as _ } from "../Tooltip/Tooltip.js";
import { clamp as h } from "../../../utils/value/number.js";
const N = ({ max: a = 1, value: t, labels: r, tooltips: n, className: i }) => {
  const c = s(() => h(0, t * 100 / a, 100), [t, a]), g = !!(r != null && r.current || r != null && r.max), y = (r == null ? void 0 : r.ariaLabel) ?? (r != null && r.current ? `${r.current}: ${t}` : `${t}/${a}`);
  return /* @__PURE__ */ e(
    "div",
    {
      role: "progressbar",
      "aria-valuenow": t,
      "aria-valuemin": 0,
      "aria-valuemax": a,
      "aria-valuetext": `${c}%`,
      "aria-label": y,
      className: u("adyen-pe-progress-bar", i),
      children: [
        /* @__PURE__ */ e("div", { className: "adyen-pe-progress-bar__track", children: [
          /* @__PURE__ */ e("div", { className: "adyen-pe-progress-bar__track-background" }),
          /* @__PURE__ */ e(
            d,
            {
              tooltipContent: n == null ? void 0 : n.progress,
              title: r == null ? void 0 : r.current,
              percentage: c,
              className: "adyen-pe-progress-bar__track-fill"
            }
          ),
          /* @__PURE__ */ e(
            d,
            {
              tooltipContent: n == null ? void 0 : n.remaining,
              title: r == null ? void 0 : r.max,
              percentage: 100 - c,
              className: "adyen-pe-progress-bar__track-remaining"
            }
          )
        ] }),
        g && /* @__PURE__ */ e("div", { className: "adyen-pe-progress-bar__legend", "aria-hidden": "true", children: [
          r.current && /* @__PURE__ */ e(o, { el: m.SPAN, variant: p.CAPTION, className: "adyen-pe-progress-bar__legend-label", children: r.current }),
          r.max && /* @__PURE__ */ e(o, { el: m.SPAN, variant: p.CAPTION, className: "adyen-pe-progress-bar__legend-label", children: r.max })
        ] })
      ]
    }
  );
}, d = ({ tooltipContent: a, title: t, percentage: r, className: n }) => {
  const i = (c) => /* @__PURE__ */ e("div", { className: n, title: c, style: { width: `${r}%` } });
  return a ? /* @__PURE__ */ e(_, { content: a, children: i() }) : i(t);
}, C = N;
export {
  C as default
};
