// @adyen/adyen-platform-experience-web/dist/es/components/internal/Calendar/calendar/facade/Calendar.js
var li = Object.defineProperty;
var fi = (m, i, e) => i in m ? li(m, i, { enumerable: !0, configurable: !0, writable: !0, value: e }) : m[i] = e;
var _t = (m, i, e) => (fi(m, typeof i != "symbol" ? i + "" : i, e), e), Ht = (m, i, e) => {
  if (!i.has(m))
    throw TypeError("Cannot " + e);
};
var t = (m, i, e) => (Ht(m, i, "read from private field"), e ? e.call(m) : i.get(m)), h = (m, i, e) => {
  if (i.has(m))
    throw TypeError("Cannot add the same private member more than once");
  i instanceof WeakSet ? i.add(m) : i.set(m, e);
}, r = (m, i, e, o) => (Ht(m, i, "write to private field"), o ? o.call(m, e) : i.set(m, e), e);
var l = (m, i, e) => (Ht(m, i, "access private method"), e);
import { InteractionKeyCode as g } from "../../../../types.js";
import { SELECT_NONE as nt, CALENDAR_CONTROLS as jt, SHIFT_BLOCK as ui, SHIFT_PERIOD as wt, SHIFT_FRAME as mi, CALENDAR_SELECTIONS as di, SELECT_MANY as Y, FRAME_SIZES as Si, FIRST_WEEK_DAYS as Ei, CURSOR_NEXT_BLOCK as Oi, CURSOR_PREV_BLOCK as Ri, CURSOR_BLOCK_END as gi, CURSOR_LINE_END as Ci, CURSOR_BLOCK_START as Ti, CURSOR_LINE_START as pi, CURSOR_DOWNWARD as ki, CURSOR_UPWARD as Fi, CURSOR_FORWARD as bi, CURSOR_BACKWARD as yi, SELECT_ONE as ct, SELECTION_FROM as O, SELECTION_TO as N, CONTROLS_NONE as Ni, CONTROLS_MINIMAL as Ai, CONTROLS_ALL as Di, SELECTION_COLLAPSE as Ii } from "../constants.js";
import { withTimezone as Wt } from "../utils.js";
import { CalendarShiftControlsFlag as Vt, CalendarShiftControlFlag as j } from "../types.js";
import Gt from "../../../../../primitives/time/today/main.js";
import { createIndexed as Mt } from "../../../../../primitives/auxiliary/indexed/main.js";
import { createWatchlist as Li, isWatchlistUnsubscribeToken as _i } from "../../../../../primitives/reactive/watchlist/main.js";
import { pickFrom as V } from "../../../../../utils/collection/main.js";
import { boolify as G, boolOrTrue as Xt } from "../../../../../utils/value/bool.js";
import { createEffectStack as qt } from "../../../../../primitives/reactive/effectStack/main.js";
import Hi from "../timeframe/frames/MonthFrame.js";
import { timezoneToSystem as lt, systemToTimezone as ft } from "../../../../../core/Localization/datetime/restamper/utils.js";
import { EMPTY_OBJECT as X } from "../../../../../utils/value/constants.js";
import { withFreezeProxyHandlers as wi, structFrom as Wi, struct as Mi } from "../../../../../utils/struct/main.js";
import { isNullish as ut, isFunction as L, isString as Jt, isUndefined as Zt } from "../../../../../utils/value/is.js";
import { noop as $t } from "../../../../../utils/common.js";
import { isBitSafeInteger as Pi } from "../../../../../utils/value/number.js";
var ti, ii, u, F, s, b, C, y, E, W, R, K, D, I, T, A, U, Z, _, p, St, H, x, M, $, Et, Ot, Rt, gt, Ct, Tt, pt, kt, si, Ft, ei, bt, hi, yt, z, mt, P, q, Q, Pt, Nt, ri, At, oi, Dt, ai, It, ni, tt, Bt, Lt, ci, B, J, it, Kt, v, dt, st, Ut, et, xt, ht, zt, rt, vt;
const f = class f {
  constructor() {
    h(this, P);
    h(this, Q);
    h(this, Nt);
    h(this, At);
    h(this, Dt);
    h(this, It);
    h(this, tt);
    h(this, Lt);
    h(this, B);
    h(this, it);
    h(this, v);
    h(this, st);
    h(this, et);
    h(this, ht);
    h(this, rt);
    _t(this, "grid");
    _t(this, "kill");
    h(this, u, X);
    h(this, F, !1);
    h(this, s, void 0);
    h(this, b, void 0);
    h(this, C, void 0);
    h(this, y, !1);
    h(this, E, nt);
    h(this, W, !1);
    h(this, R, void 0);
    h(this, K, (ti = t(this, R)) == null ? void 0 : ti.join(" "));
    h(this, D, void 0);
    h(this, I, void 0);
    h(this, T, void 0);
    h(this, A, void 0);
    h(this, U, void 0);
    h(this, Z, Gt());
    h(this, _, []);
    h(this, p, void 0);
    h(this, St, new Proxy(
      Mt(() => {
        var i;
        return ((i = t(this, p)) == null ? void 0 : i.length) ?? 0;
      }, l(this, tt, Bt).bind(this)),
      wi({
        get: (i, e, o) => {
          var a, c;
          const n = ((a = t(this, p)) == null ? void 0 : a.indexOf(e)) ?? -1;
          return n >= 0 ? (c = l(this, tt, Bt).call(this, n)) == null ? void 0 : c[1] : Reflect.get(i, e, o);
        }
      })
    ));
    h(this, H, Li({
      blocks: () => {
        var i;
        return (i = t(this, s)) == null ? void 0 : i.size;
      },
      cells: () => {
        var i;
        return (i = t(this, s)) == null ? void 0 : i.units;
      },
      controls: () => V(jt, t(this, u).controls),
      cursor: () => {
        var i;
        return (i = t(this, s)) == null ? void 0 : i.cursor;
      },
      from: () => {
        var i;
        return (i = t(this, s)) == null ? void 0 : i.selectionStart;
      },
      highlight: () => t(this, E),
      locale: () => {
        var i;
        return (i = t(this, s)) == null ? void 0 : i.locale;
      },
      minified: () => G(t(this, u).minified),
      origin: () => {
        var i;
        return (i = t(this, s)) == null ? void 0 : i.getTimestampAtIndex(0);
      },
      timezone: () => {
        var i;
        return (i = t(this, s)) == null ? void 0 : i.timezone;
      },
      to: () => {
        var i;
        return (i = t(this, s)) == null ? void 0 : i.selectionEnd;
      },
      today: () => t(this, Z).timestamp
    }));
    h(this, x, (ii = t(this, H)) == null ? void 0 : ii.snapshot);
    h(this, M, qt(() => {
      var i;
      return t(this, T) && ((i = t(this, H)) == null ? void 0 : i.requestNotification());
    }));
    h(this, $, qt(() => {
      var i;
      return (i = t(this, T)) == null ? void 0 : i.call(t(this, P, q));
    }));
    h(this, Et, Wi(
      Mt(
        () => {
          var i;
          return ((i = t(this, s)) == null ? void 0 : i.size) ?? 0;
        },
        (i) => {
          var e;
          return (e = t(this, s)) == null ? void 0 : e.frameBlocks[i];
        }
      ),
      {
        config: {
          value: Object.defineProperties(
            (i) => (i && l(this, At, oi).call(this, i), t(this, P, q)),
            {
              cursorIndex: {
                get: () => t(this, D),
                set: (i) => {
                  t(this, F) || (ut(i) ? r(this, D, void 0) : L(i) && r(this, D, i));
                }
              },
              shiftFactor: {
                get: () => t(this, I),
                set: (i) => {
                  t(this, F) || (ut(i) ? r(this, I, void 0) : L(i) && r(this, I, i));
                }
              },
              watch: {
                get: () => t(this, T),
                set: (i) => {
                  var e, o, n, a, c;
                  if (!t(this, F))
                    if (L(i)) {
                      if (r(this, T, i), !t(this, A)) {
                        const d = (e = t(this, M)) == null ? void 0 : e.bind(t(f, yt).bind(this));
                        d && (r(this, A, (o = t(this, M)) == null ? void 0 : o.bind($t)), r(this, U, (a = t(this, H)) == null ? void 0 : a.subscribe((n = t(this, $)) == null ? void 0 : n.bind(d))), t(this, s) && (t(this, s).effect = t(this, A)));
                      }
                      if (!t(this, W))
                        return;
                      r(this, W, !1), (c = t(this, A)) == null || c.call(this);
                    } else
                      ut(i) && r(this, T, void 0);
                }
              }
            }
          )
        },
        controls: { value: t(this, St) },
        cursor: {
          value: Object.defineProperties(
            (i) => l(f, z, mt).call(this, (e) => !!(e && l(this, Dt, ai).call(this, e)))(i),
            {
              valueOf: { value: () => {
                var i;
                return ((i = t(this, s)) == null ? void 0 : i.cursor) ?? -1;
              } }
            }
          )
        },
        highlight: {
          value: (() => {
            const i = () => t(this, b) === t(this, C) && Zt(t(this, C)), e = (o) => (n) => l(f, z, mt).call(this, (a) => {
              var c, d, S, k, w;
              if (!(t(this, F) || !t(this, E) || t(this, E) === nt)) {
                if (ut(a))
                  return l(this, it, Kt).call(this);
                i() ? (d = t(this, s)) == null || d.updateSelection(a, Ii) : ((c = t(this, s)) == null || c.updateSelection(a, o), t(this, E) === Y && t(this, R) && l(this, v, dt).call(this, a, o === O ? N : O, t(this, R))), r(this, b, (S = t(this, s)) == null ? void 0 : S.selectionStart), r(this, C, (k = t(this, s)) == null ? void 0 : k.selectionEnd), (w = t(this, s)) == null || w.shiftFrameToTimestamp(o === O ? t(this, b) : t(this, C));
              }
            })(n);
            return Mi({
              blank: { get: i },
              from: {
                get: () => {
                  var o;
                  return ((o = t(this, s)) == null ? void 0 : o.selectionStart) ?? t(this, b);
                },
                set: e(O)
              },
              to: {
                get: () => {
                  var o;
                  return ((o = t(this, s)) == null ? void 0 : o.selectionEnd) ?? t(this, C);
                },
                set: e(N)
              }
            });
          })()
        },
        rowspan: { get: () => {
          var i;
          return ((i = t(this, s)) == null ? void 0 : i.rowspan) ?? 0;
        } },
        weekdays: { get: () => {
          var i;
          return ((i = t(this, s)) == null ? void 0 : i.daysOfWeek) ?? t(f, gt);
        } }
      }
    ));
    this.grid = t(this, Et), this.kill = l(this, It, ni).bind(this);
  }
};
u = new WeakMap(), F = new WeakMap(), s = new WeakMap(), b = new WeakMap(), C = new WeakMap(), y = new WeakMap(), E = new WeakMap(), W = new WeakMap(), R = new WeakMap(), K = new WeakMap(), D = new WeakMap(), I = new WeakMap(), T = new WeakMap(), A = new WeakMap(), U = new WeakMap(), Z = new WeakMap(), _ = new WeakMap(), p = new WeakMap(), St = new WeakMap(), H = new WeakMap(), x = new WeakMap(), M = new WeakMap(), $ = new WeakMap(), Et = new WeakMap(), Ot = new WeakMap(), Rt = new WeakMap(), gt = new WeakMap(), Ct = new WeakMap(), Tt = new WeakMap(), pt = new WeakMap(), kt = new WeakSet(), si = function(i) {
  if (!Jt(i) || !t(f, Ot).test(i))
    return;
  const e = i.split(/\s+/);
  return Array.from({ length: 6 }, (o, n) => parseInt(e[n] ?? "0"));
}, Ft = new WeakSet(), ei = function(i) {
  switch (i & ~j.PREV) {
    case j.FRAME:
      return mi;
    case j.PERIOD:
      return wt;
    case j.BLOCK:
    default:
      return ui;
  }
}, bt = new WeakSet(), hi = function(i) {
  return i & j.PREV ? -1 : 1;
}, yt = new WeakMap(), z = new WeakSet(), mt = function(i) {
  var e;
  return ((e = t(this, M)) == null ? void 0 : e.bind(i)) ?? i;
}, P = new WeakSet(), q = function() {
  return { ...t(this, u) };
}, Q = new WeakSet(), Pt = function() {
  return new Hi();
}, Nt = new WeakSet(), ri = function(i) {
  return !!t(this, s) && !(i > 0 ? t(this, s).isAtEnd : t(this, s).isAtStart);
}, At = new WeakSet(), oi = function(i) {
  var n, a;
  if (t(this, F))
    return;
  r(this, R, void 0);
  const e = i == null ? void 0 : i.highlight, o = G(t(this, u).minified);
  if (Jt(e) ? r(this, R, l(n = f, kt, si).call(n, e)) && r(this, E, Y) : r(this, E, V(di, e, t(this, E))), r(this, u, {
    ...t(this, u),
    ...i,
    blocks: V(Si, i == null ? void 0 : i.blocks, t(this, u).blocks),
    controls: V(jt, i == null ? void 0 : i.controls, t(this, u).controls),
    firstWeekDay: V(Ei, i == null ? void 0 : i.firstWeekDay, t(this, u).firstWeekDay),
    fixedBlockHeight: G(i == null ? void 0 : i.fixedBlockHeight, t(this, u).fixedBlockHeight),
    highlight: t(this, E),
    minified: G(i == null ? void 0 : i.minified, t(this, u).minified),
    trackCurrentDay: G(i == null ? void 0 : i.trackCurrentDay, t(this, u).trackCurrentDay)
  }), !L(t(this, T))) {
    t(this, s) ? r(this, W, !0) : (r(this, s, t(this, Q, Pt)), l(this, et, xt).call(this), l(this, rt, vt).call(this), l(this, ht, zt).call(this));
    return;
  }
  (!t(this, s) || o !== t(this, u).minified) && (r(this, s, t(this, Q, Pt)), t(this, s).effect = t(this, A)), l(this, et, xt).call(this), (a = t(this, A)) == null || a.call(this);
}, Dt = new WeakSet(), ai = function(i) {
  if (i && t(this, s) && L(t(this, T))) {
    if (i instanceof KeyboardEvent) {
      switch (i.code) {
        case g.ARROW_LEFT:
          t(this, s).shiftFrameCursor(yi);
          break;
        case g.ARROW_RIGHT:
          t(this, s).shiftFrameCursor(bi);
          break;
        case g.ARROW_UP:
          t(this, s).shiftFrameCursor(Fi);
          break;
        case g.ARROW_DOWN:
          t(this, s).shiftFrameCursor(ki);
          break;
        case g.HOME:
          t(this, s).shiftFrameCursor(i.ctrlKey ? Ti : pi);
          break;
        case g.END:
          t(this, s).shiftFrameCursor(i.ctrlKey ? gi : Ci);
          break;
        case g.PAGE_UP:
          i.shiftKey ? t(this, s).shiftFrameByOffset(-1, wt) : t(this, s).shiftFrameCursor(Ri);
          break;
        case g.PAGE_DOWN:
          i.shiftKey ? t(this, s).shiftFrameByOffset(1, wt) : t(this, s).shiftFrameCursor(Oi);
          break;
        case g.SPACE:
        case g.ENTER:
          return l(this, B, J).call(this), !0;
        default:
          return;
      }
      return t(this, y) && l(this, B, J).call(this, X), !0;
    }
    if (i instanceof MouseEvent && t(f, Rt).includes(i.type) && L(t(this, D))) {
      const e = t(this, D).call(t(this, P, q), i);
      if (!Pi(e))
        return;
      const o = i.type === "click";
      if (!(o || t(this, y)))
        return;
      if (t(this, s).shiftFrameCursor(e), t(this, s).cursor === e)
        return o ? l(this, B, J).call(this) : l(this, B, J).call(this, X), !0;
    }
  }
}, It = new WeakSet(), ni = function() {
  var i;
  t(this, F) || ((i = t(this, U)) == null || i.call(this), r(this, M, r(this, $, r(this, D, r(this, s, r(this, E, r(this, K, r(this, x, r(this, R, r(this, I, r(this, U, r(this, H, r(this, A, r(this, T, void 0))))))))))))), r(this, u, X), r(this, y, r(this, W, !1)), r(this, F, !0));
}, tt = new WeakSet(), Bt = function(i) {
  var o, n;
  if (!t(this, p) || i < 0 || i >= t(this, p).length)
    return;
  const e = t(this, p)[i];
  if (!t(this, _)[i]) {
    const a = Vt[e], c = l(o = f, Ft, ei).call(o, a), d = l(n = f, bt, hi).call(n, a);
    t(this, _)[i] = (...S) => l(f, z, mt).call(this, (...k) => {
      var at;
      const w = l(this, Nt, ri).call(this, d);
      if (!(w && k.length))
        return w;
      const ot = l(this, Lt, ci).call(this, e, k[0]);
      return Zt(ot) ? !1 : ((at = t(this, s)) == null || at.shiftFrameByOffset(d * ot, c), !0);
    })(...S);
  }
  return [e, t(this, _)[i]];
}, Lt = new WeakSet(), ci = function(i, e) {
  if (!(t(this, s) && L(t(this, T))))
    return;
  if (e instanceof MouseEvent) {
    if (e.type !== "click")
      return;
  } else if (e instanceof KeyboardEvent) {
    if (!t(f, Ct).includes(e.code))
      return;
  } else
    return;
  let o = 1;
  if (L(t(this, I))) {
    const n = Number(t(this, I).call(t(this, P, q), e, i));
    o = Number.isInteger(n) && n >= 1 ? n : o;
  }
  return o;
}, B = new WeakSet(), J = function(i) {
  if (t(this, F) || !t(this, s))
    return;
  switch (t(this, E)) {
    case Y:
    case ct:
      break;
    case nt:
    default:
      return;
  }
  const e = t(this, s).cursor, o = Math.max(t(this, s).getTimestampAtIndex(e), t(this, s).timeslice.from), n = Math.min(t(this, s).getTimestampAtIndex(e + 1) - 1, t(this, s).timeslice.to), a = t(this, R);
  if (t(this, E) === ct || t(this, s).blankSelection || a)
    if (r(this, y, !(t(this, E) === ct || a)), t(this, E) === Y && a) {
      const c = n >= t(this, s).selectionEnd ? O : N;
      c === O ? t(this, s).updateSelection(n, N) : t(this, s).updateSelection(o, O), l(this, v, dt).call(this, c === O ? t(this, s).selectionEnd : t(this, s).selectionStart, c, a);
    } else
      t(this, s).updateSelection(o, O), t(this, s).updateSelection(n, N);
  else {
    const c = i === X, d = Wt(t(this, s).timezone);
    if (c || r(this, y, !1), o <= t(this, s).selectionStart) {
      const S = new Date(lt(d, t(this, s).selectionStart)), k = Math.min(
        ft(d, S.setDate(S.getDate() + 1) - 1),
        t(this, s).timeslice.to
      );
      o === t(this, s).selectionStart && n <= k && t(this, s).updateSelection(n, N), t(this, s).updateSelection(o, O);
    } else {
      const S = new Date(lt(d, t(this, s).selectionEnd)), k = Math.max(
        ft(d, S.setHours(0, 0, 0, 0)),
        t(this, s).timeslice.from
      );
      o <= t(this, s).selectionEnd && o >= k && t(this, s).updateSelection(o, O), t(this, s).updateSelection(n, N);
    }
    if (c)
      return;
  }
  r(this, b, t(this, s).selectionStart), r(this, C, t(this, s).selectionEnd);
}, it = new WeakSet(), Kt = function() {
  var i;
  (i = t(this, s)) == null || i.clearSelection(), r(this, y, !1), r(this, b, r(this, C, void 0));
}, v = new WeakSet(), dt = function(i, e, o) {
  var Yt;
  if (!t(this, s))
    return;
  const n = Wt((Yt = t(this, s)) == null ? void 0 : Yt.timezone), a = new Date(lt(n, i)), c = e === O ? -1 : 1, [d = 0, S = 0, k = 0, w = 0, ot = 0, at = 0] = o ?? [];
  a.setFullYear(
    a.getFullYear() + d * c,
    a.getMonth() + S * c,
    a.getDate() + k * c
  ), a.setHours(
    a.getHours() + w * c,
    a.getMinutes() + ot * c,
    a.getSeconds() + at * c
  ), t(this, s).updateSelection(ft(n, a.getTime() - c), e);
}, st = new WeakSet(), Ut = function() {
  var i, e;
  t(this, b) && ((i = t(this, s)) == null || i.updateSelection(t(this, b), O)), t(this, C) && ((e = t(this, s)) == null || e.updateSelection(t(this, C), N)), r(this, y, !1);
}, et = new WeakSet(), xt = function() {
  t(this, s) && (t(this, s).timeslice = t(this, u).timeslice, t(this, s).dynamicBlockHeight = !t(this, u).fixedBlockHeight, t(this, s).firstWeekDay = t(this, u).firstWeekDay, t(this, s).locale = t(this, u).locale, t(this, s).size = t(this, u).blocks, t(this, s).timezone = t(this, u).timezone, t(this, s).trackCurrentDay = t(this, u).trackCurrentDay, r(this, Z, Gt(t(this, s).timezone)), l(this, st, Ut).call(this));
}, ht = new WeakSet(), zt = function() {
  var i, e, o, n, a, c;
  switch (t(this, E)) {
    case Y:
      !Xt((i = t(this, s)) == null ? void 0 : i.blankSelection) && t(this, R) && l(this, v, dt).call(this, (e = t(this, s)) == null ? void 0 : e.selectionStart, N, t(this, R));
      break;
    case ct:
      if (!Xt((o = t(this, s)) == null ? void 0 : o.blankSelection)) {
        const d = Wt((n = t(this, s)) == null ? void 0 : n.timezone), S = new Date(lt(d, (a = t(this, s)) == null ? void 0 : a.selectionStart));
        (c = t(this, s)) == null || c.updateSelection(ft(d, S.setHours(23, 59, 59, 999)), N);
      }
      break;
    case nt:
    default:
      l(this, it, Kt).call(this);
      return;
  }
}, rt = new WeakSet(), vt = function() {
  var i, e;
  switch ((i = t(this, H)) == null ? void 0 : i.snapshot.controls) {
    case Di:
      r(this, p, t(f, Tt));
      break;
    case Ai:
      r(this, p, t(f, pt));
      break;
    case Ni:
    default:
      r(this, p, void 0);
  }
  t(this, _).length = 0, t(this, _).length = ((e = t(this, p)) == null ? void 0 : e.length) ?? 0;
}, h(f, kt), h(f, Ft), h(f, bt), h(f, z), h(f, Ot, /^(?:0|[1-9]\d*)(\s+(?:0|[1-9]\d*)?){0,5}?$/), h(f, Rt, ["click", "mouseover", "pointerover"]), h(f, gt, Mt(0, $t)), h(f, Ct, [g.ENTER, g.SPACE]), h(f, Tt, Object.keys(Vt).filter((i) => isNaN(+i))), h(f, pt, ["PREV", "NEXT"]), h(f, yt, function(i) {
  var c, d;
  if (_i(i))
    return;
  let e = !1, o = !1, n = !1;
  const a = (c = t(this, R)) == null ? void 0 : c.join(" ");
  for (const S of Object.keys(i))
    i[S] !== ((d = t(this, x)) == null ? void 0 : d[S]) && (S === "controls" ? e = !0 : S === "highlight" ? o = !0 : (S === "from" || S === "to") && (n = !0));
  t(this, K) !== a && (r(this, K, a), o = !0), r(this, x, i), t(this, y) && !n && l(this, st, Ut).call(this), e && l(this, rt, vt).call(this), o && l(this, ht, zt).call(this);
});
let Qt = f;
export {
  Qt as default
};
