// @adyen/adyen-platform-experience-web/dist/es/components/internal/FormFields/Select/components/SelectButton.js
import { jsx as a } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import T from "classnames";
import { useMemo as N } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import S from "../../../../../core/Context/useCoreContext.js";
import { ButtonVariant as C } from "../../../Button/types.js";
import f from "../../../Typography/Typography.js";
import L from "../../../SVGIcons/ChevronDown.js";
import { TypographyElement as u, TypographyVariant as B } from "../../../Typography/types.js";
import { DROPDOWN_BUTTON_CLASS as p, DROPDOWN_BUTTON_ACTIVE_CLASS as R, DROPDOWN_BUTTON_HAS_SELECTION_CLASS as h, DROPDOWN_BUTTON_READONLY_CLASS as y, DROPDOWN_BUTTON_INVALID_CLASS as A, DROPDOWN_BUTTON_VALID_CLASS as E, DROPDOWN_BUTTON_ICON_CLASS as P, DROPDOWN_BUTTON_TEXT_CLASS as U, DROPDOWN_BUTTON_MULTI_SELECT_COUNTER_CLASS as g, DROPDOWN_BUTTON_COLLAPSE_INDICATOR_CLASS as x, DROPDOWN_BUTTON_CLASSNAME as I } from "../constants.js";
import { Fragment as W } from "../../../../../external/preact/dist/preact.module.js";
import b from "../../../Img/Img.js";
import w from "../../../Button/Button.js";
import { boolOrFalse as V } from "../../../../../utils/value/bool.js";
const v = ({
  active: e,
  disabled: m,
  className: o,
  filterable: i,
  toggleButtonRef: l,
  ...n
}) => {
  const r = N(() => i ? T(I, o) : o, [o, i]);
  return i ? /* @__PURE__ */ a("div", { ...n, className: r, ref: l }) : /* @__PURE__ */ a(
    w,
    {
      ...n,
      className: r,
      disabled: m,
      variant: C.SECONDARY,
      ref: l
    }
  );
}, F = (e) => {
  var d;
  const { i18n: m } = S(), { active: o, filterable: i, multiSelect: l, placeholder: n, readonly: r, showList: s, withoutCollapseIndicator: _ } = e, O = N(() => (n == null ? void 0 : n.trim()) || m.get("select.filter.placeholder"), [m, n]), t = N(() => V(l) ? void 0 : o[0], [o, l]), D = N(() => {
    var c;
    return ((c = t == null ? void 0 : t.name) == null ? void 0 : c.trim()) || O;
  }, [t, O]);
  return /* @__PURE__ */ a(
    v,
    {
      active: o,
      disabled: r,
      "aria-disabled": r,
      "aria-expanded": s,
      "aria-haspopup": "listbox",
      className: T(p, {
        [R]: s,
        [h]: !!o.length,
        [y]: r,
        [A]: e.isInvalid,
        [E]: e.isValid
      }),
      filterable: i,
      onClick: r ? void 0 : e.toggleList,
      onKeyDown: r ? void 0 : e.onButtonKeyDown,
      role: i ? "button" : void 0,
      tabIndex: 0,
      title: D,
      toggleButtonRef: e.toggleButtonRef,
      type: i ? "" : "button",
      "aria-describedby": e.ariaDescribedBy,
      id: e.id ?? "",
      children: [
        s && i ? /* @__PURE__ */ a(
          "input",
          {
            "aria-autocomplete": "list",
            "aria-controls": e.selectListId,
            "aria-expanded": s,
            "aria-owns": e.selectListId,
            autoComplete: "off",
            className: "adyen-pe-filter-input",
            onInput: e.onInput,
            placeholder: O,
            ref: e.filterInputRef,
            role: "combobox",
            type: "text"
          }
        ) : /* @__PURE__ */ a(W, { children: [
          (t == null ? void 0 : t.icon) && /* @__PURE__ */ a(b, { className: P, src: t.icon, alt: t.name.trim() }),
          /* @__PURE__ */ a("span", { className: U, children: ((d = t == null ? void 0 : t.selectedOptionName) == null ? void 0 : d.trim()) || (t == null ? void 0 : t.name.trim()) || O }),
          l && e.appliedFilterNumber > 0 && /* @__PURE__ */ a("div", { className: g, children: /* @__PURE__ */ a(f, { el: u.SPAN, variant: B.BODY, stronger: !0, children: e.appliedFilterNumber }) })
        ] }),
        !_ && /* @__PURE__ */ a("span", { className: x, children: /* @__PURE__ */ a(L, { role: "presentation" }) })
      ]
    }
  );
}, Z = F;
export {
  Z as default
};
