// @adyen/adyen-platform-experience-web/dist/es/components/internal/DataGrid/components/TableHeaderCell.js
import { jsx as i } from "../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import o from "classnames";
import { CellTextPosition as n } from "../types.js";
import { useDataGridContext as c } from "../hooks/useDataGridContext.js";
import { useRef as s, useEffect as m } from "../../../../external/preact/hooks/dist/hooks.module.js";
const h = ({ cellKey: e, position: t, label: l }) => {
  const { registerCells: a } = c(), r = s(null);
  return m(() => {
    var d;
    r.current && a({
      column: e,
      width: (d = r.current) == null ? void 0 : d.getBoundingClientRect().width
    });
  }, [e, a]), /* @__PURE__ */ i(
    "div",
    {
      role: "columnheader",
      id: String(e),
      className: o("adyen-pe-data-grid__cell adyen-pe-data-grid__cell--heading", {
        "adyen-pe-data-grid__cell--right": t === n.RIGHT,
        "adyen-pe-data-grid__cell--center": t === n.CENTER
      }),
      children: /* @__PURE__ */ i("div", { ref: r, className: "adyen-pe-data-grid__cell--heading-content", children: l })
    }
  );
};
export {
  h as TableHeaderCell
};
