// @adyen/adyen-platform-experience-web/dist/es/components/external/TransactionDetails/components/refund/TransactionRefundReason.js
import { jsx as n } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { useMemo as f, useCallback as l } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import { TX_DATA_CONTAINER as d, TX_DATA_INPUT_HEAD as T, TX_DATA_INPUT_CONTAINER as u, TX_DATA_INPUT_CONTAINER_SHORT as R } from "../constants.js";
import { REFUND_REASONS as p } from "../../context/constants.js";
import N from "../../../../../core/Context/useCoreContext.js";
import A from "../../../../internal/Typography/Typography.js";
import { TypographyElement as _, TypographyVariant as h } from "../../../../internal/Typography/types.js";
import { useTransactionRefundContext as g } from "../../context/refund/context.js";
import C from "../../../../internal/FormFields/Select/Select.js";
const D = () => {
  const { i18n: o } = N(), { interactionsDisabled: s, refundReason: i, setRefundReason: a } = g(), m = f(
    () => Object.freeze(
      p.map((e) => ({ id: e, name: o.has(`refundReason.${e}`) ? o.get(`refundReason.${e}`) : e }))
    ),
    [o]
  ), c = l(
    (e) => {
      var t;
      const r = (t = e.target) == null ? void 0 : t.value;
      r && a(r);
    },
    [a]
  );
  return /* @__PURE__ */ n("div", { className: d, children: [
    /* @__PURE__ */ n("div", { className: T, children: /* @__PURE__ */ n(A, { el: _.SPAN, variant: h.BODY, stronger: !0, children: o.get("refundReason") }) }),
    /* @__PURE__ */ n("div", { className: `${u} ${R}`, children: /* @__PURE__ */ n(
      C,
      {
        items: m,
        readonly: s,
        filterable: !1,
        multiSelect: !1,
        onChange: c,
        selected: i
      }
    ) })
  ] });
}, P = D;
export {
  P as default
};
