// @adyen/adyen-platform-experience-web/dist/es/external/preact/compat/dist/compat.module.js
import { Component as s, options as i, createElement as p, Fragment as m, toChildArray as v, render as y } from "../../dist/preact.module.js";
import { createContext as Q } from "../../dist/preact.module.js";
import { useCallback as q, useContext as X, useEffect as ee, useImperativeHandle as te, useLayoutEffect as re, useMemo as oe, useReducer as ne, useRef as ue, useState as ae } from "../../hooks/dist/hooks.module.js";
function h(e, t) {
  for (var r in e)
    if (r !== "__source" && !(r in t))
      return !0;
  for (var o in t)
    if (o !== "__source" && e[o] !== t[o])
      return !0;
  return !1;
}
function b(e, t) {
  this.props = e, this.context = t;
}
function I(e, t) {
  function r(n) {
    var a = this.props.ref, u = a == n.ref;
    return !u && a && (a.call ? a(null) : a.current = null), t ? !t(this.props, n) || !u : h(this.props, n);
  }
  function o(n) {
    return this.shouldComponentUpdate = r, p(e, n);
  }
  return o.displayName = "Memo(" + (e.displayName || e.name) + ")", o.prototype.isReactComponent = !0, o.__f = !0, o;
}
(b.prototype = new s()).isPureReactComponent = !0, b.prototype.shouldComponentUpdate = function(e, t) {
  return h(this.props, e) || h(this.state, t);
};
var C = i.__b;
i.__b = function(e) {
  e.type && e.type.__f && e.ref && (e.props.ref = e.ref, e.ref = null), C && C(e);
};
var E = typeof Symbol < "u" && Symbol.for && Symbol.for("react.forward_ref") || 3911;
function Z(e) {
  function t(r) {
    if (!("ref" in r))
      return e(r, null);
    var o = r.ref;
    delete r.ref;
    var n = e(r, o);
    return r.ref = o, n;
  }
  return t.$$typeof = E, t.render = t, t.prototype.isReactComponent = t.__f = !0, t.displayName = "ForwardRef(" + (e.displayName || e.name) + ")", t;
}
var U = i.__e;
i.__e = function(e, t, r, o) {
  if (e.then) {
    for (var n, a = t; a = a.__; )
      if ((n = a.__c) && n.__c)
        return t.__e == null && (t.__e = r.__e, t.__k = r.__k), n.__c(e, t);
  }
  U(e, t, r, o);
};
var g = i.unmount;
function R(e, t, r) {
  return e && (e.__c && e.__c.__H && (e.__c.__H.__.forEach(function(o) {
    typeof o.__c == "function" && o.__c();
  }), e.__c.__H = null), (e = function(o, n) {
    for (var a in n)
      o[a] = n[a];
    return o;
  }({}, e)).__c != null && (e.__c.__P === r && (e.__c.__P = t), e.__c = null), e.__k = e.__k && e.__k.map(function(o) {
    return R(o, t, r);
  })), e;
}
function w(e, t, r) {
  return e && r && (e.__v = null, e.__k = e.__k && e.__k.map(function(o) {
    return w(o, t, r);
  }), e.__c && e.__c.__P === t && (e.__e && r.appendChild(e.__e), e.__c.__e = !0, e.__c.__P = r)), e;
}
function d() {
  this.__u = 0, this.t = null, this.__b = null;
}
function O(e) {
  var t = e.__.__c;
  return t && t.__a && t.__a(e);
}
function f() {
  this.u = null, this.o = null;
}
i.unmount = function(e) {
  var t = e.__c;
  t && t.__R && t.__R(), t && 32 & e.__u && (e.type = null), g && g(e);
}, (d.prototype = new s()).__c = function(e, t) {
  var r = t.__c, o = this;
  o.t == null && (o.t = []), o.t.push(r);
  var n = O(o.__v), a = !1, u = function() {
    a || (a = !0, r.__R = null, n ? n(_) : _());
  };
  r.__R = u;
  var _ = function() {
    if (!--o.__u) {
      if (o.state.__a) {
        var l = o.state.__a;
        o.__v.__k[0] = w(l, l.__c.__P, l.__c.__O);
      }
      var c;
      for (o.setState({ __a: o.__b = null }); c = o.t.pop(); )
        c.forceUpdate();
    }
  };
  o.__u++ || 32 & t.__u || o.setState({ __a: o.__b = o.__v.__k[0] }), e.then(u, u);
}, d.prototype.componentWillUnmount = function() {
  this.t = [];
}, d.prototype.render = function(e, t) {
  if (this.__b) {
    if (this.__v.__k) {
      var r = document.createElement("div"), o = this.__v.__k[0].__c;
      this.__v.__k[0] = R(this.__b, r, o.__O = o.__P);
    }
    this.__b = null;
  }
  var n = t.__a && p(m, null, e.fallback);
  return n && (n.__u &= -33), [p(m, null, t.__a ? null : e.children), n];
};
var k = function(e, t, r) {
  if (++r[1] === r[0] && e.o.delete(t), e.props.revealOrder && (e.props.revealOrder[0] !== "t" || !e.o.size))
    for (r = e.u; r; ) {
      for (; r.length > 3; )
        r.pop()();
      if (r[1] < r[0])
        break;
      e.u = r = r[2];
    }
};
function A(e) {
  return this.getChildContext = function() {
    return e.context;
  }, e.children;
}
function W(e) {
  var t = this, r = e.i;
  t.componentWillUnmount = function() {
    y(null, t.l), t.l = null, t.i = null;
  }, t.i && t.i !== r && t.componentWillUnmount(), t.l || (t.i = r, t.l = { nodeType: 1, parentNode: r, childNodes: [], contains: function() {
    return !0;
  }, appendChild: function(o) {
    this.childNodes.push(o), t.i.appendChild(o);
  }, insertBefore: function(o, n) {
    this.childNodes.push(o), t.i.appendChild(o);
  }, removeChild: function(o) {
    this.childNodes.splice(this.childNodes.indexOf(o) >>> 1, 1), t.i.removeChild(o);
  } }), y(p(A, { context: t.context }, e.__v), t.l);
}
function G(e, t) {
  var r = p(W, { __v: e, i: t });
  return r.containerInfo = t, r;
}
(f.prototype = new s()).__a = function(e) {
  var t = this, r = O(t.__v), o = t.o.get(e);
  return o[0]++, function(n) {
    var a = function() {
      t.props.revealOrder ? (o.push(n), k(t, e, o)) : n();
    };
    r ? r(a) : a();
  };
}, f.prototype.render = function(e) {
  this.u = null, this.o = /* @__PURE__ */ new Map();
  var t = v(e.children);
  e.revealOrder && e.revealOrder[0] === "b" && t.reverse();
  for (var r = t.length; r--; )
    this.o.set(t[r], this.u = [1, 0, this.u]);
  return e.children;
}, f.prototype.componentDidUpdate = f.prototype.componentDidMount = function() {
  var e = this;
  this.o.forEach(function(t, r) {
    k(e, r, t);
  });
};
var H = typeof Symbol < "u" && Symbol.for && Symbol.for("react.element") || 60103, $ = /^(?:accent|alignment|arabic|baseline|cap|clip(?!PathU)|color|dominant|fill|flood|font|glyph(?!R)|horiz|image(!S)|letter|lighting|marker(?!H|W|U)|overline|paint|pointer|shape|stop|strikethrough|stroke|text(?!L)|transform|underline|unicode|units|v|vector|vert|word|writing|x(?!C))[A-Z]/, L = /^on(Ani|Tra|Tou|BeforeInp|Compo)/, M = /[A-Z0-9]/g, V = typeof document < "u", B = function(e) {
  return (typeof Symbol < "u" && typeof Symbol() == "symbol" ? /fil|che|rad/ : /fil|che|ra/).test(e);
};
s.prototype.isReactComponent = {}, ["componentWillMount", "componentWillReceiveProps", "componentWillUpdate"].forEach(function(e) {
  Object.defineProperty(s.prototype, e, { configurable: !0, get: function() {
    return this["UNSAFE_" + e];
  }, set: function(t) {
    Object.defineProperty(this, e, { configurable: !0, writable: !0, value: t });
  } });
});
var x = i.event;
function F() {
}
function j() {
  return this.cancelBubble;
}
function D() {
  return this.defaultPrevented;
}
i.event = function(e) {
  return x && (e = x(e)), e.persist = F, e.isPropagationStopped = j, e.isDefaultPrevented = D, e.nativeEvent = e;
};
var T = { enumerable: !1, configurable: !0, get: function() {
  return this.class;
} }, N = i.vnode;
i.vnode = function(e) {
  typeof e.type == "string" && function(t) {
    var r = t.props, o = t.type, n = {}, a = o.indexOf("-") === -1;
    for (var u in r) {
      var _ = r[u];
      if (!(u === "value" && "defaultValue" in r && _ == null || V && u === "children" && o === "noscript" || u === "class" || u === "className")) {
        var l = u.toLowerCase();
        u === "defaultValue" && "value" in r && r.value == null ? u = "value" : u === "download" && _ === !0 ? _ = "" : l === "translate" && _ === "no" ? _ = !1 : l[0] === "o" && l[1] === "n" ? l === "ondoubleclick" ? u = "ondblclick" : l !== "onchange" || o !== "input" && o !== "textarea" || B(r.type) ? l === "onfocus" ? u = "onfocusin" : l === "onblur" ? u = "onfocusout" : L.test(u) && (u = l) : l = u = "oninput" : a && $.test(u) ? u = u.replace(M, "-$&").toLowerCase() : _ === null && (_ = void 0), l === "oninput" && n[u = l] && (u = "oninputCapture"), n[u] = _;
      }
    }
    o == "select" && n.multiple && Array.isArray(n.value) && (n.value = v(r.children).forEach(function(c) {
      c.props.selected = n.value.indexOf(c.props.value) != -1;
    })), o == "select" && n.defaultValue != null && (n.value = v(r.children).forEach(function(c) {
      c.props.selected = n.multiple ? n.defaultValue.indexOf(c.props.value) != -1 : n.defaultValue == c.props.value;
    })), r.class && !r.className ? (n.class = r.class, Object.defineProperty(n, "className", T)) : (r.className && !r.class || r.class && r.className) && (n.class = n.className = r.className), t.props = n;
  }(e), e.$$typeof = H, N && N(e);
};
var P = i.__r;
i.__r = function(e) {
  P && P(e), e.__c;
};
var S = i.diffed;
i.diffed = function(e) {
  S && S(e);
  var t = e.props, r = e.__e;
  r != null && e.type === "textarea" && "value" in t && t.value !== r.value && (r.value = t.value == null ? "" : t.value);
};
export {
  s as Component,
  m as Fragment,
  b as PureComponent,
  d as Suspense,
  f as SuspenseList,
  Q as createContext,
  p as createElement,
  G as createPortal,
  Z as forwardRef,
  I as memo,
  q as useCallback,
  X as useContext,
  ee as useEffect,
  te as useImperativeHandle,
  re as useLayoutEffect,
  oe as useMemo,
  ne as useReducer,
  ue as useRef,
  ae as useState
};
