// @adyen/adyen-platform-experience-web/dist/es/components/external/CapitalOffer/components/utils/CapitalErrorMessageDisplay.js
import { jsx as o } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { getCapitalErrorMessage as s } from "../../../../utils/capital/getCapitalErrorMessage.js";
import p from "../../../../../core/Errors/AdyenPlatformExperienceError.js";
import { ErrorMessageDisplay as l } from "../../../../internal/ErrorMessageDisplay/ErrorMessageDisplay.js";
import { useCallback as f, useMemo as u } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import E from "../../../../internal/Button/Button.js";
import { ButtonVariant as c } from "../../../../internal/Button/types.js";
import C from "../../../../../core/Context/useCoreContext.js";
import { ErrorTypes as g } from "../../../../../core/Http/utils.js";
import { Fragment as d } from "../../../../../external/preact/dist/preact.module.js";
const A = ({
  emptyGrantOffer: t,
  error: e,
  onContactSupport: a,
  onBack: r
}) => {
  const { i18n: i } = C(), m = f(
    () => /* @__PURE__ */ o(d, { children: r && /* @__PURE__ */ o(E, { variant: c.SECONDARY, onClick: r, children: i.get("back") }) }),
    [i, r]
  ), n = u(() => t ? new p(g.ERROR, "EmptyConfig", "Empty Configuration", "EMPTY_CONFIG") : e, [t, e]);
  return /* @__PURE__ */ o(
    l,
    {
      absolutePosition: !1,
      withImage: !0,
      renderSecondaryButton: m,
      outlined: !1,
      ...s(n, a)
    }
  );
};
export {
  A as CapitalErrorMessageDisplay
};
