// @adyen/adyen-platform-experience-web/dist/es/components/internal/Calendar/hooks/useCalendarControlsRendering.js
import { jsx as l } from "../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { ButtonVariant as m } from "../../Button/types.js";
import { useCallback as c } from "../../../../external/preact/hooks/dist/hooks.module.js";
import f from "../../SVGIcons/ChevronDown.js";
import d from "../../SVGIcons/ChevronUp.js";
import p from "../../../../core/Context/useCoreContext.js";
import C from "../../../../hooks/element/useDetachedRender.js";
import R from "../../Button/Button.js";
import { isFunction as b } from "../../../../utils/value/is.js";
const h = (e) => {
  const { i18n: a } = p();
  return C(
    c(
      (u, r, t) => {
        if (!(u instanceof HTMLElement))
          return null;
        if (b(e))
          return e(r, t);
        let o, n, i;
        switch (r) {
          case "PREV":
            o = "prev", n = "previous", i = d;
            break;
          case "NEXT":
            o = n = "next", i = f;
            break;
          default:
            return null;
        }
        const s = t();
        return s ? /* @__PURE__ */ l(
          R,
          {
            "aria-label": a.get(`calendar.${n}Month`),
            variant: m.TERTIARY,
            disabled: !s,
            classNameModifiers: ["circle", o],
            iconButton: !0,
            onClick: t,
            children: /* @__PURE__ */ l(i, { role: "presentation" })
          },
          r
        ) : null;
      },
      [a, e]
    )
  );
}, D = h;
export {
  D as default
};
