// @adyen/adyen-platform-experience-web/dist/es/components/external/CapitalOverview/components/PreQualified/PreQualified.js
import { jsx as e } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import S from "../PreQualifiedIntro.js";
import { useState as g, useCallback as t, useMemo as T } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import { CapitalOffer as x } from "../../../CapitalOffer/components/CapitalOffer/CapitalOffer.js";
import { Fragment as O } from "../../../../../external/preact/dist/preact.module.js";
const j = ({
  hideTitle: m,
  dynamicOffer: n,
  skipPreQualifiedIntro: o,
  onOfferOptionsRequest: r,
  onFundsRequest: p,
  onOfferDismiss: a
}) => {
  const [l, c] = g(!!o), f = t(() => c(!0), []), u = t(() => {
    r ? r() : f();
  }, [f, r]), s = t(() => c(!1), []), C = t(() => {
    a ? a() : s();
  }, [s, a]), i = T(() => o || l ? "CapitalOffer" : "Intro", [l, o]);
  return /* @__PURE__ */ e(O, { children: i === "Intro" ? /* @__PURE__ */ e(S, { hideTitle: m, dynamicOffer: n, onSeeOptions: u }) : i === "CapitalOffer" ? /* @__PURE__ */ e(x, { onFundsRequest: p, onOfferDismiss: C, externalDynamicOffersConfig: n }) : null });
};
export {
  j as PreQualified
};
