// @adyen/adyen-platform-experience-web/dist/es/index.js
import "./components/shared.scss.js";
import "./style/index.scss.js";
import r from "./core/core.js";
import { all_locales as p, da_DK as n, de_DE as m, en_US as _, es_ES as u, fi_FI as d, fr_FR as x, it_IT as c, nl_NL as v, no_NO as w, pt_BR as O, sv_SE as y } from "./translations/index.js";
import { default as E } from "./components/external/PayoutDetails/PayoutDetails.js";
import { default as P } from "./components/external/PayoutsOverview/PayoutsOverviewElement.js";
import { default as S } from "./components/external/TransactionDetails/TransactionDetails.js";
import { default as F } from "./components/external/TransactionsOverview/TransactionsOverviewElement.js";
import { default as N } from "./components/external/CapitalOffer/CapitalOfferElement.js";
import { default as A } from "./components/external/ReportsOverview/ReportsOverviewElement.js";
import { default as K } from "./components/external/CapitalOverview/CapitalOverviewElement.js";
async function i(e) {
  return await new r(e).initialize();
}
export {
  i as AdyenPlatformExperience,
  N as CapitalOffer,
  K as CapitalOverview,
  r as Core,
  E as PayoutDetails,
  P as PayoutsOverview,
  A as ReportsOverview,
  S as TransactionDetails,
  F as TransactionsOverview,
  p as all_locales,
  n as da_DK,
  m as de_DE,
  _ as en_US,
  u as es_ES,
  d as fi_FI,
  x as fr_FR,
  c as it_IT,
  v as nl_NL,
  w as no_NO,
  O as pt_BR,
  y as sv_SE
};
