// @adyen/adyen-platform-experience-web/dist/es/components/external/CapitalOffer/components/CapitalOfferSummary/CapitalOfferSummary.js
import { jsx as r } from "../../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import f from "../../../../../core/Context/useCoreContext.js";
import { useMemo as c, useCallback as v } from "../../../../../external/preact/hooks/dist/hooks.module.js";
import { getExpectedRepaymentDate as x, calculateMaximumRepaymentPeriodInMonths as I, getPaymentRatePercentage as M } from "../utils/utils.js";
import n from "../../../../internal/Typography/Typography.js";
import { TypographyElement as m, TypographyVariant as l } from "../../../../internal/Typography/types.js";
import "./CapitalOfferSummary.scss.js";
import y from "../../../../internal/Button/Button.js";
import { ButtonVariant as P } from "../../../../internal/Button/types.js";
import k from "../../../../../hooks/useMutation/useMutation.js";
import { Tooltip as D } from "../../../../internal/Tooltip/Tooltip.js";
import { AlertTypeOption as E } from "../../../../internal/Alert/types.js";
import b from "../../../../internal/Alert/Alert.js";
import { CapitalErrorMessageDisplay as F } from "../utils/CapitalErrorMessageDisplay.js";
import L from "classnames";
import { CAPITAL_REPAYMENT_FREQUENCY as C } from "../../../../constants.js";
import S from "../../../../internal/InfoBox/InfoBox.js";
import W from "../../../../internal/StructuredList/StructuredList.js";
import q from "../../../../internal/Icon/Icon.js";
import { useAuthContext as Y } from "../../../../../core/Auth/context.js";
import { EMPTY_OBJECT as w } from "../../../../../utils/value/constants.js";
const H = ["30_013"], le = ({
  grantOffer: t,
  onBack: h,
  onFundsRequest: u,
  onContactSupport: p
}) => {
  const { i18n: e } = f(), R = c(() => {
    const a = x(t.expectedRepaymentPeriodDays);
    return a ? e.date(a, { month: "long" }) : null;
  }, [t, e]), { requestFunds: N } = Y().endpoints, o = k({
    queryFn: N,
    options: {
      onSuccess: (a) => u == null ? void 0 : u(a)
    }
  }), A = v(
    (a) => {
      o.mutate(w, { path: { grantOfferId: a } });
    },
    [o]
  ), T = v(() => {
    t.id && A(t.id);
  }, [t.id, A]), s = c(
    () => I(t.maximumRepaymentPeriodDays),
    [t.maximumRepaymentPeriodDays]
  ), i = c(() => {
    const a = o.error ? o.error : null;
    if (a && H.includes(a.errorCode))
      switch (a.errorCode) {
        case "30_013":
          return {
            title: e.get("capital.thereIsNoPrimaryAccountConfigured"),
            message: e.get("capital.weCouldNotContinueWithTheOfferContactSupportForHelp"),
            errorCode: "30_013"
          };
        default:
          return {
            title: e.get("somethingWentWrong"),
            message: e.get("capital.weCouldNotLoadFinancialOffers")
          };
      }
    return null;
  }, [e, o.error]), _ = c(() => {
    const a = [
      {
        key: "capital.fees",
        value: e.amount(t.feesAmount.value, t.feesAmount.currency)
      },
      {
        key: "capital.totalRepaymentAmount",
        value: e.amount(t.totalAmount.value, t.totalAmount.currency)
      },
      {
        key: "capital.repaymentThreshold",
        value: e.amount(t.thresholdAmount.value, t.thresholdAmount.currency)
      },
      {
        key: "capital.dailyRepaymentRate",
        value: e.get("capital.xPercent", { values: { percentage: M(t.repaymentRate) } })
      },
      {
        key: "capital.expectedRepaymentPeriod",
        value: e.get("capital.xDays", { values: { days: t.expectedRepaymentPeriodDays } })
      },
      { key: "account", value: e.get("capital.primaryAccount") }
    ];
    return s && a.splice(4, 0, {
      key: "capital.maximumRepaymentPeriod",
      value: s === 1 ? e.get("capital.oneMonth") : e.get("capital.xMonths", { values: { months: s } })
    }), a;
  }, [t, e, s]);
  return !i && o.error ? /* @__PURE__ */ r(F, { error: o.error, onBack: h, onContactSupport: p }) : /* @__PURE__ */ r("div", { className: "adyen-pe-capital-offer-summary", children: [
    /* @__PURE__ */ r(S, { className: "adyen-pe-capital-offer-summary__grant-summary", children: [
      /* @__PURE__ */ r(n, { el: m.PARAGRAPH, variant: l.BODY, children: [
        e.get("capital.youAreRequestingFundingOf"),
        " ",
        /* @__PURE__ */ r("strong", { children: `${e.amount(t.grantAmount.value, t.grantAmount.currency, { minimumFractionDigits: 0 })}.` })
      ] }),
      /* @__PURE__ */ r(n, { el: m.PARAGRAPH, variant: l.CAPTION, children: e.get("capital.youWillNeedToRepayAMinimumOfXEveryXDaysToPayOffTheFunds", {
        values: {
          amount: e.amount(t.thresholdAmount.value, t.thresholdAmount.currency),
          days: C,
          date: R ?? ""
        }
      }) })
    ] }),
    /* @__PURE__ */ r(
      W,
      {
        classNames: "adyen-pe-capital-offer-summary__details",
        renderLabel: (a, d) => d === "capital.repaymentThreshold" ? /* @__PURE__ */ r(
          D,
          {
            isContainerHovered: !0,
            content: e.get("capital.minimumRepaymentToRepayTheFinancingOnTime", {
              values: { days: C }
            }),
            children: /* @__PURE__ */ r("span", { children: /* @__PURE__ */ r(
              n,
              {
                className: "adyen-pe-capital-offer-summary__list-label",
                el: m.SPAN,
                variant: l.CAPTION,
                children: a
              }
            ) })
          }
        ) : /* @__PURE__ */ r(
          n,
          {
            className: "adyen-pe-capital-offer-summary__list-label",
            el: m.SPAN,
            variant: l.CAPTION,
            children: a
          }
        ),
        renderValue: (a, d) => {
          const g = d === "account" && o.error && i && i.errorCode === "30_013";
          return /* @__PURE__ */ r(
            n,
            {
              className: L({
                "adyen-pe-capital-offer-summary__details--error": g
              }),
              el: m.SPAN,
              variant: l.CAPTION,
              stronger: !0,
              children: [
                g ? /* @__PURE__ */ r(q, { name: "warning-filled" }) : null,
                a
              ]
            }
          );
        },
        items: _
      }
    ),
    i && /* @__PURE__ */ r(
      b,
      {
        className: "adyen-pe-capital-offer-summary__error-alert",
        type: E.WARNING,
        title: i.title,
        description: i.message,
        children: p ? /* @__PURE__ */ r(y, { className: "adyen-pe-capital-offer-summary__error-alert-button", onClick: p, children: e.get("contactSupport") }) : null
      }
    ),
    /* @__PURE__ */ r("div", { className: "adyen-pe-capital-offer-summary__buttons", children: [
      o.error && !i ? null : /* @__PURE__ */ r(y, { variant: P.SECONDARY, onClick: h, children: e.get("back") }),
      /* @__PURE__ */ r(
        y,
        {
          variant: P.PRIMARY,
          state: o.isLoading ? "loading" : void 0,
          onClick: T,
          disabled: o.isLoading || !!o.error,
          children: e.get(o.isLoading ? "capital.requesting" : "capital.requestFunds")
        }
      )
    ] })
  ] });
};
export {
  le as CapitalOfferSummary
};
