// @adyen/adyen-platform-experience-web/dist/es/components/external/PayoutDetails/components/PayoutData.js
import { jsx as n } from "../../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import O from "classnames";
import { useMemo as N } from "../../../../external/preact/hooks/dist/hooks.module.js";
import H from "../../../../core/Context/useCoreContext.js";
import F from "../../../internal/Accordion/Accordion.js";
import _ from "../../../internal/Card/Card.js";
import { TypographyVariant as e } from "../../../internal/Typography/types.js";
import m from "../../../internal/Typography/Typography.js";
import z from "../../../internal/DataOverviewDetails/DataOverviewDetailsSkeleton.js";
import G from "../../../../hooks/useTimezoneAwareDateFormatting.js";
import "./PayoutData.scss.js";
import { PD_BASE_CLASS as J, PD_TITLE_CLASS as K, PD_SECTION_CLASS as l, PD_TITLE_BA_CLASS as k, PD_CONTENT_CLASS as q, PD_CARD_CLASS as S, PD_SECTION_AMOUNT_CLASS as f, PD_SECTION_GROSS_AMOUNT_CLASS as R, PD_CARD_TITLE_CLASS as $, PD_SECTION_NET_AMOUNT_CLASS as Q, PD_UNPAID_AMOUNT as W } from "./constants.js";
import { DATE_FORMAT_PAYOUT_DETAILS as X } from "../../../../constants/dateFormats.js";
import y from "../../../internal/StructuredList/StructuredList.js";
import { Fragment as Z } from "../../../../external/preact/dist/preact.module.js";
import { EMPTY_OBJECT as b } from "../../../../utils/value/constants.js";
const Ci = ({
  balanceAccountId: u,
  balanceAccountDescription: p,
  payout: s,
  isFetching: j
}) => {
  var B;
  const { payout: t } = s ?? b, { dateFormat: P } = G("UTC"), { i18n: r } = H(), o = N(() => {
    var a, h;
    const c = (h = (a = s == null ? void 0 : s.amountBreakdowns) == null ? void 0 : a.adjustmentBreakdown) == null ? void 0 : h.reduce(
      (d, i) => {
        var L, I, E, D, Y, U, M;
        const v = (L = i == null ? void 0 : i.amount) != null && L.value && ((I = i == null ? void 0 : i.amount) != null && I.currency) ? r.amount((E = i == null ? void 0 : i.amount) == null ? void 0 : E.value, (D = i == null ? void 0 : i.amount) == null ? void 0 : D.currency, { hideCurrency: !0 }) : (((Y = i == null ? void 0 : i.amount) == null ? void 0 : Y.value) ?? "").toString(), A = `${i == null ? void 0 : i.category}`, g = r.get(A), C = i != null && i.category && g !== A ? g : i == null ? void 0 : i.category;
        return i != null && i.category && v && C && d[(U = i == null ? void 0 : i.amount) != null && U.value && ((M = i == null ? void 0 : i.amount) == null ? void 0 : M.value) < 0 ? "subtractions" : "additions"].push({ key: C, value: v }), d;
      },
      { subtractions: [], additions: [] }
    );
    return c == null || c.subtractions.sort((d, i) => d.key.localeCompare(i.key)), c == null || c.additions.sort((d, i) => d.key.localeCompare(i.key)), c;
  }, [r, s]), T = N(() => {
    var a, h;
    const c = (h = (a = s == null ? void 0 : s.amountBreakdowns) == null ? void 0 : a.fundsCapturedBreakdown) == null ? void 0 : h.reduce((d, i) => {
      var v, A, g, C;
      return ((v = i == null ? void 0 : i.amount) == null ? void 0 : v.value) === 0 || (A = i == null ? void 0 : i.amount) != null && A.value && i.category && d.push({
        key: i.category,
        value: r.amount((g = i == null ? void 0 : i.amount) == null ? void 0 : g.value, (C = i == null ? void 0 : i.amount) == null ? void 0 : C.currency, { hideCurrency: !0 })
      }), d;
    }, []);
    return c == null || c.sort((d, i) => d.key === "capture" ? -1 : i.key === "capture" ? 1 : d.key.localeCompare(i.key)), c;
  }, [s, r]), x = N(
    () => t != null && t.createdAt ? P(new Date(t == null ? void 0 : t.createdAt), X) : "",
    [t, P]
  );
  return /* @__PURE__ */ n(Z, { children: t ? /* @__PURE__ */ n("div", { className: J, children: [
    /* @__PURE__ */ n("div", { className: K, children: [
      /* @__PURE__ */ n(m, { variant: e.SUBTITLE, stronger: !0, children: r.get("netPayout") }),
      /* @__PURE__ */ n(m, { variant: e.TITLE, large: !0, children: `${r.amount(t.payoutAmount.value, t.payoutAmount.currency, {
        hideCurrency: !0
      })} ${t.payoutAmount.currency}` }),
      /* @__PURE__ */ n(m, { variant: e.BODY, children: x }),
      /* @__PURE__ */ n("div", { className: l, children: [
        p && /* @__PURE__ */ n(m, { variant: e.CAPTION, stronger: !0, wide: !0, children: `${p}` }),
        /* @__PURE__ */ n(m, { variant: e.CAPTION, className: k, children: `${u}` })
      ] })
    ] }),
    /* @__PURE__ */ n("div", { className: q, children: [
      /* @__PURE__ */ n("div", { className: l, children: (t == null ? void 0 : t.fundsCapturedAmount) && (T && Object.keys(T).length > 0 ? /* @__PURE__ */ n(
        F,
        {
          header: /* @__PURE__ */ n(m, { variant: e.BODY, children: r.get("fundsCaptured") }),
          headerInformation: /* @__PURE__ */ n(m, { variant: e.BODY, children: r.amount(t.fundsCapturedAmount.value, t.fundsCapturedAmount.currency) }),
          children: /* @__PURE__ */ n("div", { className: l, children: /* @__PURE__ */ n("div", { className: S, children: /* @__PURE__ */ n(_, { noPadding: !0, children: /* @__PURE__ */ n(y, { items: T }) }) }) })
        }
      ) : /* @__PURE__ */ n("div", { className: O(f, R), children: [
        /* @__PURE__ */ n(m, { variant: e.BODY, children: r.get("fundsCaptured") }),
        /* @__PURE__ */ n(m, { variant: e.BODY, children: r.amount(t.fundsCapturedAmount.value, t.fundsCapturedAmount.currency) })
      ] })) }),
      /* @__PURE__ */ n("div", { className: l, children: o != null && o.subtractions && Object.keys(o == null ? void 0 : o.subtractions).length > 0 || o != null && o.additions && Object.keys(o == null ? void 0 : o.additions).length > 0 ? /* @__PURE__ */ n(
        F,
        {
          header: /* @__PURE__ */ n(m, { variant: e.BODY, children: r.get("adjustments") }),
          headerInformation: /* @__PURE__ */ n(m, { variant: e.BODY, children: r.amount(t.adjustmentAmount.value, t.adjustmentAmount.currency) }),
          children: [
            (o == null ? void 0 : o.additions) && Object.keys(o == null ? void 0 : o.additions).length > 0 && /* @__PURE__ */ n("div", { className: S, children: /* @__PURE__ */ n(
              _,
              {
                noPadding: !0,
                renderHeader: /* @__PURE__ */ n(m, { className: $, variant: e.CAPTION, stronger: !0, children: r.get("additions") }),
                children: /* @__PURE__ */ n(y, { items: o == null ? void 0 : o.additions })
              }
            ) }),
            (o == null ? void 0 : o.subtractions) && Object.keys(o == null ? void 0 : o.subtractions).length > 0 && /* @__PURE__ */ n("div", { className: S, children: /* @__PURE__ */ n(
              _,
              {
                noPadding: !0,
                renderHeader: /* @__PURE__ */ n(m, { className: $, variant: e.CAPTION, stronger: !0, children: r.get("subtractions") }),
                children: /* @__PURE__ */ n(y, { items: o == null ? void 0 : o.subtractions })
              }
            ) })
          ]
        }
      ) : /* @__PURE__ */ n("div", { className: O(f, R), children: [
        /* @__PURE__ */ n(m, { variant: e.BODY, children: r.get("adjustments") }),
        /* @__PURE__ */ n(m, { variant: e.BODY, children: r.amount(t.adjustmentAmount.value, t.adjustmentAmount.currency) })
      ] }) }),
      /* @__PURE__ */ n("div", { className: O(l), children: /* @__PURE__ */ n("div", { className: O(f, Q), children: [
        /* @__PURE__ */ n(m, { variant: e.BODY, stronger: !0, children: r.get("netPayout") }),
        /* @__PURE__ */ n(m, { variant: e.BODY, stronger: !0, children: r.amount(t.payoutAmount.value, t.payoutAmount.currency) })
      ] }) })
    ] }),
    ((B = s == null ? void 0 : s.payout) == null ? void 0 : B.unpaidAmount) && /* @__PURE__ */ n("div", { className: W, children: [
      /* @__PURE__ */ n(m, { variant: e.BODY, children: r.get("remainingAmount") }),
      /* @__PURE__ */ n(m, { variant: e.BODY, children: r.amount(s.payout.unpaidAmount.value, s.payout.unpaidAmount.currency) })
    ] })
  ] }) : /* @__PURE__ */ n(z, { skeletonRowNumber: 6 }) });
};
export {
  Ci as PayoutData
};
