// @adyen/adyen-platform-experience-web/dist/es/components/internal/Modal/Modal.js
import { jsx as e } from "../../../external/preact/jsx-runtime/dist/jsxRuntime.module.js";
import { useClickOutside as h } from "../../../hooks/element/useClickOutside.js";
import { useResponsiveViewport as v, mediaQueries as _ } from "../../../hooks/useResponsiveViewport.js";
import { ButtonVariant as x } from "../Button/types.js";
import g from "../../../core/Context/useCoreContext.js";
import { useContext as E, useCallback as k, useEffect as N } from "../../../external/preact/hooks/dist/hooks.module.js";
import { createContext as C, Fragment as M } from "../../../external/preact/dist/preact.module.js";
import m from "classnames";
import "./Modal.scss.js";
import b from "../Button/Button.js";
import B from "../Icon/Icon.js";
const i = C({ withinModal: !1 }), P = () => E(i);
function Q({
  title: n,
  children: c,
  classNameModifiers: p = [],
  isOpen: a,
  onClose: o,
  isDismissible: d = !0,
  headerWithBorder: s = !0,
  size: r = "fluid",
  ...u
}) {
  const f = v(_.down.xs), { i18n: y } = g(), w = h(null, o), t = k(
    (l) => {
      l.key === "Escape" && a && d && o();
    },
    [a, d, o]
  );
  return N(() => (a ? window.addEventListener("keydown", t) : window.removeEventListener("keydown", t), () => window.removeEventListener("keydown", t)), [a, t]), /* @__PURE__ */ e(M, { children: a && /* @__PURE__ */ e(
    "div",
    {
      className: m(
        "adyen-pe-modal-wrapper",
        p.map((l) => `adyen-pe-modal-wrapper--${l}`),
        { "adyen-pe-modal-wrapper--open": a, "adyen-pe-modal-wrapper--dismissible": d }
      ),
      role: "dialog",
      "aria-modal": "true",
      "aria-hidden": !open,
      ...u,
      children: /* @__PURE__ */ e(i.Provider, { value: { withinModal: !0 }, children: /* @__PURE__ */ e(
        "div",
        {
          className: m("adyen-pe-modal", {
            "adyen-pe-modal--fluid": r === "fluid",
            "adyen-pe-modal--small": r === "small",
            "adyen-pe-modal--large": r === "large",
            "adyen-pe-modal--extra-large": r === "extra-large",
            "adyen-pe-modal--full-screen": r === "full-screen" || f
          }),
          ref: w,
          children: [
            /* @__PURE__ */ e(
              "div",
              {
                className: m("adyen-pe-modal__header", {
                  "adyen-pe-modal__header--with-title": n,
                  "adyen-pe-modal__header--with-border-bottom": s
                }),
                children: [
                  n && /* @__PURE__ */ e("div", { className: "adyen-pe-modal__header__title", children: n }),
                  d && /* @__PURE__ */ e(
                    b,
                    {
                      onClick: o,
                      variant: x.TERTIARY,
                      iconButton: !0,
                      classNameModifiers: ["circle"],
                      className: "adyen-pe-modal__header-icon",
                      "aria-label": y.get("dismiss"),
                      children: /* @__PURE__ */ e(B, { name: "cross" })
                    }
                  )
                ]
              }
            ),
            /* @__PURE__ */ e("div", { className: "adyen-pe-modal__content", children: c })
          ]
        }
      ) })
    }
  ) });
}
export {
  Q as default,
  P as useModalContext
};
